{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Schistosoma Mansoni RNA Seq analysis of Intestinal and Liver Gene Expression Levels\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Introduction"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Schistosoma Mansoni is a parasitic flatworm that infects snails and humans. When the schistosoma mansoni infects its human host, it matures and chooses a final destination to reproduce in, either the liver or the intestines. This infection is the cause of the tropical disease, Schistosomiasis. Schistosomiasis has a high global prevalence, around 700 million people are at risk of infection and 200,000 deaths occur annually. [(stat for death)](https://pmc.ncbi.nlm.nih.gov/articles/PMC6997417/#:~:text=Two%20hundred%20and%20thirty%20million,increased%20sensitivity%20to%20pathological%20antigens.) Most research done to design drugs and develop treatment for this disease focuses on analysis of the schistosoma that reside in the host's liver. The liver schistosoma are not only easiers to study, but they also cause more of the symptoms and damage. However, the intestinal schistosoma are responsible for the continuation of the species. The eggs from the intestinal worms are excreted and hatch in freshwater, infect a snail, mature and infect a new unsuspecting host. The eggs of the liver worm are far less likely to escape from the liver, when the host immune system recognizes an egg it will surround the egg. This forms a granuloma, granulomas lead to fibrosis which can lead to severe chronic liver disease. Any worms that don't make it to the intestines will remain trapped in the host until they die. [(liver eggs life)](https://pmc.ncbi.nlm.nih.gov/articles/PMC7815046/#:~:text=Summary,feces%20of%20mice%20and%20humans.) [(granuloma)](https://pmc.ncbi.nlm.nih.gov/articles/PMC8951358/#:~:text=Through%20a%20complex%20migration%20process,a%20therapeutic%20target%2C%20are%20discussed.) To develop a treatment for this disease, we analyze the genes expressed by the schistosoma to determine potential targets for drug intervention.Although a drug that targets a liver worm may reduce or eliminate symptoms, the eggs and worms that make it to the intestines are still able to infect a new host. If worms found anywhere in the host express the same genes then the current research focusing on liver worms will apply to liver worms as well. However, if gene expression levels are significantly different in the different worms then there is a need for instestinal worm analysis research to ensure treatments will be effective against all worms in the host. To determine if gene expression differs significantly between liver and intestinal worms, RNA sequencing and analysis was used to analyze and compare samples from both the liver and the intestines of mouse hosts. Discoveries made in this study will inform the direction of future research. Results from the RNAseq analysis can validate previous research and uncover unique interactions in the host's intestinal environment. This study aims to be reproducible and use popular computational approaches in transcriptomic analysis to contribute to bioinformatics.\n",
    "<!-- overview of disease and worm life, why we study gene expression?, why we compare gene expression between liver and Intestinal  -->\n",
    "\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Methods\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Data Access"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To promote and demonstrate reproducibility, this study utilized data publicly available from a previous study conducted by Peterková et al. According to the study's methods, Schistosoma mansoni egg samples were harvested from the liver and instestines of experimentally infected mice, in accordance with approved animal welfare protocols. \n",
    "\n",
    "\n",
    "<!-- [idk what this is but it looks somewhat useful](https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/) -->"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Sequencing and Library Prep"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Following collection, mature and immature eggs were separated, incubated overnight in [RNAlater Stabilization Solution](https://www.thermofisher.com/us/en/home/brands/product-brand/rnalater.html?ef_id=CjwKCAjwiezABhBZEiwAEbTPGIZGfMdpb1xjBA0ZpqDHCduax2C1Hc_4Q0DNuMAAqJkioaeza5_EFxoCAm0QAvD_BwE:G:s&s_kwcid=AL!3652!3!596199967644!e!!g!!rnalater%20solution!15924075301!133607799058&cid=bid_sap_rst_r01_co_cp1362_pjt0000_bid00000_0se_gaw_bt_pur_con&gad_source=1&gad_campaignid=15924075301&gbraid=0AAAAADxi_GQU2FBNrH80g0UUIB1zDocRt&gclid=CjwKCAjwiezABhBZEiwAEbTPGIZGfMdpb1xjBA0ZpqDHCduax2C1Hc_4Q0DNuMAAqJkioaeza5_EFxoCAm0QAvD_BwE) and stored at -80°C. The eggs were removed from the RNAlater, placed in [Trizol](https://www.thermofisher.com/us/en/home/brands/product-brand/trizol.html?ef_id=CjwKCAjwiezABhBZEiwAEbTPGDCsQT-Y-DLdaJzo6T6CL6oyXBB9IJt0-Nz6nef_lH5Irofcy9WhqRoCIBoQAvD_BwE:G:s&s_kwcid=AL!3652!3!614256677213!e!!g!!trizol!15924075301!138911523839&cid=bid_sap_rst_r01_co_cp1362_pjt0000_bid00000_0se_gaw_bt_pur_con&gad_source=1&gad_campaignid=15924075301&gbraid=0AAAAADxi_GQU2FBNrH80g0UUIB1zDocRt&gclid=CjwKCAjwiezABhBZEiwAEbTPGDCsQT-Y-DLdaJzo6T6CL6oyXBB9IJt0-Nz6nef_lH5Irofcy9WhqRoCIBoQAvD_BwE) and mechanically homogenized using a [plastic pestle](https://www.usascientific.com/pestle-microcentrifuge-tubes/p/Pestle-Micro). Total RNA was isolated and treated with TURBO DNase[(Invitrogen)](https://www.thermofisher.com/order/catalog/product/AM2238?ef_id=CjwKCAjwiezABhBZEiwAEbTPGKoWLRYZ-zfGlITUPNfVP6a-2LYmJi9wNqZznE9919e4MJhBg2Ro8BoCMXMQAvD_BwE:G:s&s_kwcid=AL!3652!3!737048274237!e!!g!!turbo%20dnase!10552817723!176101660556&cid=bid_mol_clo_r01_co_cp1358_pjt0000_bid00000_0se_gaw_bt_pur_con&gad_source=1&gad_campaignid=10552817723&gbraid=0AAAAADxi_GTcaY8JiKLFiyCxQKQoV4GgC&gclid=CjwKCAjwiezABhBZEiwAEbTPGKoWLRYZ-zfGlITUPNfVP6a-2LYmJi9wNqZznE9919e4MJhBg2Ro8BoCMXMQAvD_BwE), and quantified using NanoDrop Spectrophotometer[(Thermo Scientific)](https://www.thermofisher.com/us/en/home/industrial/spectroscopy-elemental-isotope-analysis/molecular-spectroscopy/uv-vis-spectrophotometry/instruments/nanodrop.html). To ensure data quality despite potential RNA degredation, cDNA libraries were prepared from 250ng of total RNA using the [QuantSeq FWD 3'](https://www.lexogen.com/store/quantseq-3-mrna-seq-v2-fwd-with-udi/) mRNA Library Prep Kit (Lexogen) with [UMI Second Strand Syntesis Module for QuantSeq FWD](https://www.lexogen.com/store/umi-second-strand-synthesis-module-for-quantseq-fwd/?utm_term=&utm_campaign=DSA+Campaign&utm_source=adwords&utm_medium=ppc&hsa_acc=1206466263&hsa_cam=1046909256&hsa_grp=56840592852&hsa_ad=691577386736&hsa_src=g&hsa_tgt=dsa-19959388920&hsa_kw=&hsa_mt=&hsa_net=adwords&hsa_ver=3&gad_source=1&gad_campaignid=1046909256&gbraid=0AAAAADtpWl8sJcj-Rjq8qgSF5LrCabEsS&gclid=CjwKCAjwiezABhBZEiwAEbTPGG7WmbArXnMqa0lOcV-6SId_XcMrmrGM5PhEdQwDBMfLi1HzWpaEkBoCiRAQAvD_BwE) and Lexogen i4 6 nt Unique Dual Indexing Add-on Kit (Lexogen). Library quality was assessed with [QuantiFluor](https://www.promega.com/products/rna-analysis/dna-and-rna-quantitation/quantifluor-dsdna-system/?catNum=E2671) dsDNA System [(Promega)](https://www.promega.com/resources/protocols/technical-manuals/101/quantifluor-dsdna-system-protocol/) and [High Sensitivity NGS Fragment](https://www.agilent.com/store/en_US/Prod-DNF-474-0500/DNF-474-0500&utm_source=google&utm_medium=free-listing&utm_campaign=shopping-org&utm_term=DNF-474-0500?gQT=1) Analysis Kit [(Agilent Technologies)](https://www.agilent.com/cs/library/usermanuals/public/quick-guide-dnf-474-hs-ngs-fragment-kit-SD-AT000134.pdf). Sequencing was performed on an Illumina NexSeq 500 (Illumina) using High Output [Kit v2.5](https://www.illumina.com/products/by-type/sequencing-kits/cluster-gen-sequencing-reagents/nextseq-series-kits-v2-5.html) 75 Cycles in single-end mode and produced approximately 10 million reads per sample. The resulting FASTQ files containing the reads were uploaded to [NCBI GenBank Database with BioProject assession number PRJNA1026419.](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1036419)\n",
    "\n",
    "\n",
    "<!-- Following collection, mature and immature eggs were separated, incubated overnight in RNAlater Stabilization Solution and stored at -80°C. The eggs were removed from the RNAlater, placed in Trizol and mechanically homogenized using a plastic pestle. Total RNA was isolated and treated with TURBO DNase (Invitrogen), and quantified using a NanoDrop Spectrophotometer (Thermo Scientific). To ensure quality despite potential degradation of RNA, cDNA libraries were prepared from 250 ng of total RNA using the QuantSeq FWD 3' mRNA Library Prep Kit (Lexogen) with UMI Second Strand Synthesis Module for QuantSeq FWD and Lexogen i4 6 nt Unique Dual Indexing Add-on Kit (Lexogen). Library quality was assessed with QuantiFluor dsDNA System (Promega) and High Sensitivity NGS Fragment Analysis Kit (Agilent Technologies). Sequencing was performed on an Illumina NexSeq 500 (Illumina) using High Output Kit v2.5 75 Cycles in single-end mode and produced approximately 10 million reads per sample. The resulting FASTQ files containing the reads were uploaded to the NCBI GenBank Database with BioProject accession number PRJNA1036419. -->"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[QuantSeq 3’ mRNA-Seq V2 Library Prep Kit FWD with UDI](https://www.lexogen.com/store/quantseq-3-mrna-seq-v2-fwd-with-udi/) \n",
    "\n",
    "[UMI Second Strand Synthesis Module for QuantSeq FWD (Illumina, Read 1)](https://www.lexogen.com/store/umi-second-strand-synthesis-module-for-quantseq-fwd/) \n",
    "\n",
    "[New NextSeq 550](https://www.illumina.com/systems/sequencing-platforms/nextseq.html)  -\n",
    "[order](https://www.illumina.com/systems/sequencing-platforms/nextseq/order-nextseq-550.html) "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Sample reads QC"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To access the data from the NCBI Database for this study, the SRA run table containing the metadata and accession numbers was downloaded using the RunSelector. Using the assession numbers from the SRA table, parallel-fastq-dump from sra-tools(v3.1.1), and a bash while loop the FASTQ files were downloaded and stored in a new directory for future access. To assess the sequencing read quality, avoid bias in analysis,  and identify potenial issues such as adapter conamination, low-quality bases, and overrepresented sequences, quality control was performed using FastQC(v.0.12.1) and summarized with MultiQC(v1.17). After quality analysis, cutadapt(v4.9) was used to trim PolyA tails left from using illumina reads with polyA selection, remove 10 bases from the 3' end to adjust for NextSeq adapters, and filter out any reads shorter than 20 bases. FastQC was run on the trimmed reads and combined with raw read reports in a single multiqc report to assess quality improvement. \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "<!-- \n",
    "\n",
    "\n",
    "Provided that the data contained single-end Illumina reads generated with PolyA selection and the study didn't mention adapter trimming status, trimming was necessary to remove polyA tails, potential adapters, and any reads that were flagged during quality control. Once cutadapt trimmed the reads \n",
    "[fastqc](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)                    0.12.1\n",
    "[sra-tools](https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit)                 3.1.1\n",
    "[multiqc](https://github.com/MultiQC/MultiQC)                   1.17\n",
    "[cutadapt](https://cutadapt.readthedocs.io/en/stable/)                  4.9   \n",
    "The data was accessed through NCBI's BioProject, [(PRJNA1036419)](https://www.ncbi.nlm.nih.gov/Traces/study/?query_key=2&WebEnv=MCID_681bc5d78c0274d2f3e43571&o=acc_s%3Aa), and the SRA run table for all samples was downloaded using the RunSelector. -->\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### STAR Alignment"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reads were then mapped to to the Schistosoma mansoni reference genome using STAR aligner(v2.7.10b), a high-performance, splice-aware aligner, that was also used by peterova et al. to continue to demonstrate repeatibility. The [reference genome](https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.genomic.fa.gz ) and its [annotations](https://ftp.ebi.ac.uk/pub/databases/wormbase/parasite/releases/WBPS19/species/schistosoma_mansoni/PRJEA36577/schistosoma_mansoni.PRJEA36577.WBPS19.canonical_geneset.gtf.gz) were obtained from the [WormBase ParaSite](https://parasite.wormbase.org/Schistosoma_mansoni_prjea36577/Info/Index/). Alignment ensures that the processed reads are accurately mapped to their corresponding genomic locations, enabling presise quantification and identification of gene expression. Alignment was done using the 2-pass mode [(section 8 STAR Manual)](https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_notes/STARmanual.pdf) and 32 threads to optimize accuracy, sensitivity, and alignment efficiency. For the first path, the default parameters were used. The splice junction files from all samples were used to inform the second-pass via the `--sjdbFileChrStartEnd` option. This increases senitivity by allowing STAR to treat novel junctions as known splice sites. Input FASTQ files were decompressed using `zcat`, and  output was sorted in BAM format (`--outSAMtype BAM SortedByCoordinate`). Additional read attributes (`NH HI AS nM  RG`) were included to enhance downstream processing and annotation. Unmapped reads were retained for completeness(`--outSAMunmapped Within`). \n",
    "\n",
    "\n",
    "<!-- \n",
    "\n",
    "[STAR aligner](https://pubmed.ncbi.nlm.nih.gov/23104886/)\n",
    "(Dobin A, Davis CA, Schlesinger F, et al (2012))                      2.7.10b         -->"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Identifing differentially expressed genes"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The BAM files output by STAR alignment were then indexed using samtools(v1.6) for quick access to mapped reads. STAR mapping statistics were generated using `samtools stats` to evaluate the alignment quality. Qualimap(v2.3) was used to further assess the aligment using `bamqc` for BAM quality control `bamqc` and `rnaseq` for RNA-seq analysis. These tools validated the accuracy of the alignment and produced insights into gene expression and genome coverage. Duplicate reads were identified and removed using Picard MarkDuplicates(v3.2.0). Deduplicated reads were counted using subread's(v2.0.8) `featureCounts` to assign reads to assign reads to exons, genes, promoters, etc. The `featureCounts` tool was given the BAM output file from the mark duplicates step with parameters for paired-end reads, strand specificity and the inclustion of multi-mapped reads. Also featureCounts was configured to count reads by read group using `--byReadGroup` and to output the results in a tab-separted file to facilitate vizualiztion of results. \n",
    "<!-- [subread](https://subread.sourceforge.net/)                   2.0.8 \n",
    "[picard](https://github.com/broadinstitute/picard)                    3.2.0 \n",
    "[qualimap](http://qualimap.conesalab.org/doc_html/index.html)                  2.3 \n",
    "[samtools](https://github.com/samtools/samtools)                  1.6  -->"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Results"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Discussion "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What does it mean that we found significant difference in gene expression levels between liver and intestinal worms?\n",
    "\n",
    "Why might this be?\n",
    "\n",
    "What should we do to confirm or study this?\n",
    "\n",
    "why should we do more investigating?"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### References"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": [
    "#\n",
    "# Name                    Version                   Build  Channel\n",
    "_libgcc_mutex             0.1                 conda_forge    conda-forge\n",
    "_openmp_mutex             4.5                       2_gnu    conda-forge\n",
    "_r-mutex                  1.0.1               anacondar_1    conda-forge\n",
    "alsa-lib                  1.2.8                h166bdaf_0    conda-forge\n",
    "amply                     0.1.6              pyhd8ed1ab_0    conda-forge\n",
    "anndata                   0.10.8             pyhd8ed1ab_0    conda-forge\n",
    "anyio                     4.4.0              pyhd8ed1ab_0    conda-forge\n",
    "appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge\n",
    "argcomplete               3.5.1              pyhd8ed1ab_0    conda-forge\n",
    "argon2-cffi               23.1.0             pyhd8ed1ab_0    conda-forge\n",
    "argon2-cffi-bindings      21.2.0          py312h98912ed_4    conda-forge\n",
    "argparse-dataclass        2.0.0              pyhd8ed1ab_0    conda-forge\n",
    "array-api-compat          1.7.1              pyhd8ed1ab_0    conda-forge\n",
    "arrow                     1.3.0              pyhd8ed1ab_0    conda-forge\n",
    "asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge\n",
    "async-lru                 2.0.4              pyhd8ed1ab_0    conda-forge\n",
    "attmap                    0.13.2             pyhd8ed1ab_0    conda-forge\n",
    "attrs                     23.2.0             pyh71513ae_0    conda-forge\n",
    "babel                     2.14.0             pyhd8ed1ab_0    conda-forge\n",
    "beautifulsoup4            4.12.3             pyha770c72_0    conda-forge\n",
    "binutils_impl_linux-64    2.40                 ha1999f0_7    conda-forge\n",
    "bioconductor-annotate     1.76.0            r42hdfd78af_0    bioconda\n",
    "bioconductor-annotationdbi 1.60.0            r42hdfd78af_0    bioconda\n",
    "bioconductor-biobase      2.58.0            r42ha9d7317_1    bioconda\n",
    "bioconductor-biocgenerics 0.44.0            r42hdfd78af_0    bioconda\n",
    "bioconductor-biocparallel 1.32.5            r42hf17093f_1    bioconda\n",
    "bioconductor-biostrings   2.66.0            r42ha9d7317_1    bioconda\n",
    "bioconductor-data-packages 20231203             hdfd78af_0    bioconda\n",
    "bioconductor-delayedarray 0.24.0            r42ha9d7317_1    bioconda\n",
    "bioconductor-deseq2       1.38.0            r42hf17093f_1    bioconda\n",
    "bioconductor-genefilter   1.80.0            r42ha1e849b_1    bioconda\n",
    "bioconductor-geneplotter  1.76.0            r42hdfd78af_0    bioconda\n",
    "bioconductor-genomeinfodb 1.34.9            r42hdfd78af_0    bioconda\n",
    "bioconductor-genomeinfodbdata 1.2.9             r42hdfd78af_0    bioconda\n",
    "bioconductor-genomicranges 1.50.0            r42ha9d7317_1    bioconda\n",
    "bioconductor-iranges      2.32.0            r42ha9d7317_1    bioconda\n",
    "bioconductor-keggrest     1.38.0            r42hdfd78af_0    bioconda\n",
    "bioconductor-matrixgenerics 1.10.0            r42hdfd78af_0    bioconda\n",
    "bioconductor-s4vectors    0.36.0            r42ha9d7317_1    bioconda\n",
    "bioconductor-summarizedexperiment 1.28.0            r42hdfd78af_0    bioconda\n",
    "bioconductor-xvector      0.38.0            r42ha9d7317_1    bioconda\n",
    "bioconductor-zlibbioc     1.44.0            r42ha9d7317_2    bioconda\n",
    "bleach                    6.1.0              pyhd8ed1ab_0    conda-forge\n",
    "brotli                    1.1.0                hd590300_1    conda-forge\n",
    "brotli-bin                1.1.0                hd590300_1    conda-forge\n",
    "brotli-python             1.1.0           py312h30efb56_1    conda-forge\n",
    "bwidget                   1.9.14               ha770c72_1    conda-forge\n",
    "bzip2                     1.0.8                hd590300_5    conda-forge\n",
    "c-ares                    1.28.1               hd590300_0    conda-forge\n",
    "ca-certificates           2024.7.4             hbcca054_0    conda-forge\n",
    "cached-property           1.5.2                hd8ed1ab_1    conda-forge\n",
    "cached_property           1.5.2              pyha770c72_1    conda-forge\n",
    "cairo                     1.16.0               hb05425b_5  \n",
    "certifi                   2024.7.4           pyhd8ed1ab_0    conda-forge\n",
    "cffi                      1.16.0          py312hf06ca03_0    conda-forge\n",
    "charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge\n",
    "click                     8.1.7           unix_pyh707e725_0    conda-forge\n",
    "coin-or-cbc               2.10.11              h56f689f_0    conda-forge\n",
    "coin-or-cgl               0.60.7               h516709c_0    conda-forge\n",
    "coin-or-clp               1.17.8               h1ee7a9c_0    conda-forge\n",
    "coin-or-osi               0.108.10             haf5fa05_0    conda-forge\n",
    "coin-or-utils             2.11.11              hee58242_0    conda-forge\n",
    "coincbc                   2.10.11           0_metapackage    conda-forge\n",
    "colorama                  0.4.6              pyhd8ed1ab_0    conda-forge\n",
    "coloredlogs               15.0.1             pyhd8ed1ab_3    conda-forge\n",
    "colormath                 3.0.0                      py_2    conda-forge\n",
    "comm                      0.2.2              pyhd8ed1ab_0    conda-forge\n",
    "conda-inject              1.3.2              pyhd8ed1ab_0    conda-forge\n",
    "configargparse            1.7                pyhd8ed1ab_0    conda-forge\n",
    "connection_pool           0.0.3              pyhd3deb0d_0    conda-forge\n",
    "contourpy                 1.2.1           py312h8572e83_0    conda-forge\n",
    "cryptography              42.0.8          py312hbcc2302_0    conda-forge\n",
    "curl                      7.86.0               h2283fc2_1    conda-forge\n",
    "cutadapt                  4.9             py312hf67a6ed_0    bioconda\n",
    "cycler                    0.12.1             pyhd8ed1ab_0    conda-forge\n",
    "datrie                    0.8.2           py312h98912ed_7    conda-forge\n",
    "debugpy                   1.8.2           py312h7070661_0    conda-forge\n",
    "decorator                 5.1.1              pyhd8ed1ab_0    conda-forge\n",
    "defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge\n",
    "deprecated                1.2.14             pyh1a96a4e_0    conda-forge\n",
    "dnaio                     1.2.1           py312hf67a6ed_1    bioconda\n",
    "docutils                  0.21.2             pyhd8ed1ab_0    conda-forge\n",
    "dpath                     2.2.0              pyha770c72_0    conda-forge\n",
    "eido                      0.2.2              pyhd8ed1ab_0    conda-forge\n",
    "entrypoints               0.4                pyhd8ed1ab_0    conda-forge\n",
    "exceptiongroup            1.2.0              pyhd8ed1ab_2    conda-forge\n",
    "executing                 2.0.1              pyhd8ed1ab_0    conda-forge\n",
    "expat                     2.6.2                h59595ed_0    conda-forge\n",
    "fastqc                    0.12.1               hdfd78af_0    bioconda\n",
    "font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge\n",
    "font-ttf-inconsolata      3.000                h77eed37_0    conda-forge\n",
    "font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge\n",
    "font-ttf-ubuntu           0.83                 h77eed37_2    conda-forge\n",
    "fontconfig                2.14.2               h14ed4e7_0    conda-forge\n",
    "fonts-conda-ecosystem     1                             0    conda-forge\n",
    "fonts-conda-forge         1                             0    conda-forge\n",
    "fonttools                 4.53.0          py312h9a8786e_0    conda-forge\n",
    "fqdn                      1.5.1              pyhd8ed1ab_0    conda-forge\n",
    "freetype                  2.12.1               h267a509_2    conda-forge\n",
    "fribidi                   1.0.10               h36c2ea0_0    conda-forge\n",
    "future                    1.0.0              pyhd8ed1ab_0    conda-forge\n",
    "gcc_impl_linux-64         14.1.0               h3c94d91_0    conda-forge\n",
    "gettext                   0.22.5               h59595ed_2    conda-forge\n",
    "gettext-tools             0.22.5               h59595ed_2    conda-forge\n",
    "gffread                   0.12.7               hdcf5f25_4    bioconda\n",
    "gfortran_impl_linux-64    14.1.0               he4a1faa_0    conda-forge\n",
    "giflib                    5.2.2                hd590300_0    conda-forge\n",
    "gitdb                     4.0.11             pyhd8ed1ab_0    conda-forge\n",
    "gitpython                 3.1.43             pyhd8ed1ab_0    conda-forge\n",
    "glib                      2.74.1               h6239696_0    conda-forge\n",
    "glib-tools                2.74.1               h6239696_0    conda-forge\n",
    "graphite2                 1.3.13            h59595ed_1003    conda-forge\n",
    "gsl                       2.7                  he838d99_0    conda-forge\n",
    "gxx_impl_linux-64         14.1.0               h2879b86_0    conda-forge\n",
    "h11                       0.14.0             pyhd8ed1ab_0    conda-forge\n",
    "h2                        4.1.0              pyhd8ed1ab_0    conda-forge\n",
    "h5py                      3.11.0          py312h34c39bb_0  \n",
    "harfbuzz                  6.0.0                h8e241bc_0    conda-forge\n",
    "hdf5                      1.12.1          nompi_h4df4325_104    conda-forge\n",
    "hpack                     4.0.0              pyh9f0ad1d_0    conda-forge\n",
    "htslib                    1.10                 h244ad75_0    bioconda\n",
    "httpcore                  1.0.5              pyhd8ed1ab_0    conda-forge\n",
    "httpx                     0.27.0             pyhd8ed1ab_0    conda-forge\n",
    "humanfriendly             10.0               pyhd8ed1ab_6    conda-forge\n",
    "humanize                  4.10.0             pyhd8ed1ab_0    conda-forge\n",
    "hyperframe                6.0.1              pyhd8ed1ab_0    conda-forge\n",
    "icu                       70.1                 h27087fc_0    conda-forge\n",
    "idna                      3.7                pyhd8ed1ab_0    conda-forge\n",
    "immutables                0.20            py312h98912ed_1    conda-forge\n",
    "importlib-metadata        8.0.0              pyha770c72_0    conda-forge\n",
    "importlib_metadata        8.0.0                hd8ed1ab_0    conda-forge\n",
    "importlib_resources       6.4.0              pyhd8ed1ab_0    conda-forge\n",
    "iniconfig                 2.0.0              pyhd8ed1ab_0    conda-forge\n",
    "ipykernel                 6.29.5             pyh3099207_0    conda-forge\n",
    "ipython                   8.26.0             pyh707e725_0    conda-forge\n",
    "ipywidgets                8.1.3              pyhd8ed1ab_0    conda-forge\n",
    "isa-l                     2.31.0               hd590300_1    conda-forge\n",
    "isoduration               20.11.0            pyhd8ed1ab_0    conda-forge\n",
    "jedi                      0.19.1             pyhd8ed1ab_0    conda-forge\n",
    "jinja2                    3.1.4              pyhd8ed1ab_0    conda-forge\n",
    "joblib                    1.4.2              pyhd8ed1ab_0    conda-forge\n",
    "jpeg                      9e                   h166bdaf_2    conda-forge\n",
    "jq                        1.7.1                hd590300_0    conda-forge\n",
    "json5                     0.9.25             pyhd8ed1ab_0    conda-forge\n",
    "jsonpointer               3.0.0           py312h7900ff3_0    conda-forge\n",
    "jsonschema                4.22.0             pyhd8ed1ab_0    conda-forge\n",
    "jsonschema-specifications 2023.12.1          pyhd8ed1ab_0    conda-forge\n",
    "jsonschema-with-format-nongpl 4.22.0             pyhd8ed1ab_0    conda-forge\n",
    "jupyter                   1.0.0             pyhd8ed1ab_10    conda-forge\n",
    "jupyter-lsp               2.2.5              pyhd8ed1ab_0    conda-forge\n",
    "jupyter_client            8.6.2              pyhd8ed1ab_0    conda-forge\n",
    "jupyter_console           6.6.3              pyhd8ed1ab_0    conda-forge\n",
    "jupyter_core              5.7.2           py312h7900ff3_0    conda-forge\n",
    "jupyter_events            0.10.0             pyhd8ed1ab_0    conda-forge\n",
    "jupyter_server            2.14.1             pyhd8ed1ab_0    conda-forge\n",
    "jupyter_server_terminals  0.5.3              pyhd8ed1ab_0    conda-forge\n",
    "jupyterlab                4.2.3              pyhd8ed1ab_0    conda-forge\n",
    "jupyterlab_pygments       0.3.0              pyhd8ed1ab_1    conda-forge\n",
    "jupyterlab_server         2.27.2             pyhd8ed1ab_0    conda-forge\n",
    "jupyterlab_widgets        3.0.11             pyhd8ed1ab_0    conda-forge\n",
    "kernel-headers_linux-64   2.6.32              he073ed8_17    conda-forge\n",
    "keyutils                  1.6.1                h166bdaf_0    conda-forge\n",
    "kiwisolver                1.4.5           py312h8572e83_1    conda-forge\n",
    "krb5                      1.19.3               h08a2579_0    conda-forge\n",
    "lcms2                     2.14                 h6ed2654_0    conda-forge\n",
    "ld_impl_linux-64          2.40                 hf3520f5_7    conda-forge\n",
    "lerc                      4.0.0                h27087fc_0    conda-forge\n",
    "libaec                    1.1.3                h59595ed_0    conda-forge\n",
    "libasprintf               0.22.5               h661eb56_2    conda-forge\n",
    "libasprintf-devel         0.22.5               h661eb56_2    conda-forge\n",
    "libblas                   3.9.0           22_linux64_openblas    conda-forge\n",
    "libbrotlicommon           1.1.0                hd590300_1    conda-forge\n",
    "libbrotlidec              1.1.0                hd590300_1    conda-forge\n",
    "libbrotlienc              1.1.0                hd590300_1    conda-forge\n",
    "libcblas                  3.9.0           22_linux64_openblas    conda-forge\n",
    "libcups                   2.3.3                h3e49a29_2    conda-forge\n",
    "libcurl                   7.86.0               h2283fc2_1    conda-forge\n",
    "libdeflate                1.14                 h166bdaf_0    conda-forge\n",
    "libedit                   3.1.20191231         he28a2e2_2    conda-forge\n",
    "libev                     4.33                 hd590300_2    conda-forge\n",
    "libexpat                  2.6.2                h59595ed_0    conda-forge\n",
    "libffi                    3.4.2                h7f98852_5    conda-forge\n",
    "libgcc-devel_linux-64     14.1.0             h5d3d1c9_100    conda-forge\n",
    "libgcc-ng                 14.1.0               h77fa898_0    conda-forge\n",
    "libgettextpo              0.22.5               h59595ed_2    conda-forge\n",
    "libgettextpo-devel        0.22.5               h59595ed_2    conda-forge\n",
    "libgfortran-ng            14.1.0               h69a702a_0    conda-forge\n",
    "libgfortran5              14.1.0               hc5f4f2c_0    conda-forge\n",
    "libglib                   2.74.1               h7a41b64_0    conda-forge\n",
    "libgomp                   14.1.0               h77fa898_0    conda-forge\n",
    "libiconv                  1.17                 hd590300_2    conda-forge\n",
    "libjpeg-turbo             2.1.4                h166bdaf_0    conda-forge\n",
    "liblapack                 3.9.0           22_linux64_openblas    conda-forge\n",
    "liblapacke                3.9.0           22_linux64_openblas    conda-forge\n",
    "libnghttp2                1.58.0               h47da74e_1    conda-forge\n",
    "libnsl                    2.0.1                hd590300_0    conda-forge\n",
    "libopenblas               0.3.27          pthreads_h413a1c8_0    conda-forge\n",
    "libpng                    1.6.43               h2797004_0    conda-forge\n",
    "libsanitizer              14.1.0               hcba0ae0_0    conda-forge\n",
    "libsodium                 1.0.18               h36c2ea0_1    conda-forge\n",
    "libsqlite                 3.46.0               hde9e2c9_0    conda-forge\n",
    "libssh2                   1.11.0               h0841786_0    conda-forge\n",
    "libstdcxx-devel_linux-64  14.1.0             h5d3d1c9_100    conda-forge\n",
    "libstdcxx-ng              14.1.0               hc0a3c3a_0    conda-forge\n",
    "libtiff                   4.4.0                h82bc61c_5    conda-forge\n",
    "libuuid                   2.38.1               h0b41bf4_0    conda-forge\n",
    "libwebp-base              1.4.0                hd590300_0    conda-forge\n",
    "libxcb                    1.15                 h0b41bf4_0    conda-forge\n",
    "libxcrypt                 4.4.36               hd590300_1    conda-forge\n",
    "libxml2                   2.9.14               h22db469_4    conda-forge\n",
    "libzlib                   1.2.13               h4ab18f5_6    conda-forge\n",
    "logmuse                   0.2.6              pyh8c360ce_0    conda-forge\n",
    "lzstring                  1.0.4                   py_1001    conda-forge\n",
    "make                      4.3                  hd18ef5c_1    conda-forge\n",
    "markdown                  3.6                pyhd8ed1ab_0    conda-forge\n",
    "markdown-it-py            3.0.0              pyhd8ed1ab_0    conda-forge\n",
    "markupsafe                2.1.5           py312h98912ed_0    conda-forge\n",
    "matplotlib-base           3.9.2           py312h854627b_0    conda-forge\n",
    "matplotlib-inline         0.1.7              pyhd8ed1ab_0    conda-forge\n",
    "mdurl                     0.1.2              pyhd8ed1ab_0    conda-forge\n",
    "mistune                   3.0.2              pyhd8ed1ab_0    conda-forge\n",
    "multiqc                   1.17               pyhdfd78af_1    bioconda\n",
    "munkres                   1.1.4              pyh9f0ad1d_0    conda-forge\n",
    "natsort                   8.4.0              pyhd8ed1ab_0    conda-forge\n",
    "nbclient                  0.10.0             pyhd8ed1ab_0    conda-forge\n",
    "nbconvert                 7.16.4               hd8ed1ab_1    conda-forge\n",
    "nbconvert-core            7.16.4             pyhd8ed1ab_1    conda-forge\n",
    "nbconvert-pandoc          7.16.4               hd8ed1ab_1    conda-forge\n",
    "nbformat                  5.10.4             pyhd8ed1ab_0    conda-forge\n",
    "ncbi-vdb                  3.1.1                h4ac6f70_0    bioconda\n",
    "ncurses                   6.5                  h59595ed_0    conda-forge\n",
    "nest-asyncio              1.6.0              pyhd8ed1ab_0    conda-forge\n",
    "networkx                  3.3                pyhd8ed1ab_1    conda-forge\n",
    "notebook                  7.2.1              pyhd8ed1ab_0    conda-forge\n",
    "notebook-shim             0.2.4              pyhd8ed1ab_0    conda-forge\n",
    "numpy                     1.26.4          py312heda63a1_0    conda-forge\n",
    "oniguruma                 6.9.9                hd590300_0    conda-forge\n",
    "openjdk                   17.0.3               h58dac75_5    conda-forge\n",
    "openjpeg                  2.5.0                h7d73246_1    conda-forge\n",
    "openssl                   3.3.1                h4ab18f5_1    conda-forge\n",
    "ossuuid                   1.6.2             hf484d3e_1000    conda-forge\n",
    "overrides                 7.7.0              pyhd8ed1ab_0    conda-forge\n",
    "packaging                 24.1               pyhd8ed1ab_0    conda-forge\n",
    "pandas                    2.2.2           py312h1d6d2e6_1    conda-forge\n",
    "pandoc                    3.2.1                ha770c72_0    conda-forge\n",
    "pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge\n",
    "pango                     1.50.14              hd33c08f_0    conda-forge\n",
    "parallel-fastq-dump       0.6.7              pyhdfd78af_0    bioconda\n",
    "parso                     0.8.4              pyhd8ed1ab_0    conda-forge\n",
    "pbzip2                    1.1.13               h1fcc475_2    conda-forge\n",
    "pcre2                     10.37                hc3806b6_1    conda-forge\n",
    "peppy                     0.40.2             pyhd8ed1ab_0    conda-forge\n",
    "perl                      5.32.1          7_hd590300_perl5    conda-forge\n",
    "perl-alien-build          2.48            pl5321hec16e2b_0    bioconda\n",
    "perl-alien-libxml2        0.17            pl5321hec16e2b_0    bioconda\n",
    "perl-business-isbn        3.007           pl5321hdfd78af_0    bioconda\n",
    "perl-business-isbn-data   20210112.006    pl5321hdfd78af_0    bioconda\n",
    "perl-capture-tiny         0.48            pl5321hdfd78af_2    bioconda\n",
    "perl-carp                 1.38            pl5321hdfd78af_4    bioconda\n",
    "perl-constant             1.33            pl5321hdfd78af_2    bioconda\n",
    "perl-data-dumper          2.183           pl5321hec16e2b_1    bioconda\n",
    "perl-encode               3.19            pl5321hec16e2b_1    bioconda\n",
    "perl-exporter             5.72            pl5321hdfd78af_2    bioconda\n",
    "perl-extutils-makemaker   7.70            pl5321hd8ed1ab_0    conda-forge\n",
    "perl-ffi-checklib         0.28            pl5321hdfd78af_0    bioconda\n",
    "perl-file-chdir           0.1010          pl5321hdfd78af_3    bioconda\n",
    "perl-file-path            2.18            pl5321hd8ed1ab_0    conda-forge\n",
    "perl-file-temp            0.2304          pl5321hd8ed1ab_0    conda-forge\n",
    "perl-file-which           1.24            pl5321hd8ed1ab_0    conda-forge\n",
    "perl-importer             0.026           pl5321hdfd78af_0    bioconda\n",
    "perl-mime-base64          3.16            pl5321hec16e2b_2    bioconda\n",
    "perl-parent               0.236           pl5321hdfd78af_2    bioconda\n",
    "perl-path-tiny            0.122           pl5321hdfd78af_0    bioconda\n",
    "perl-pathtools            3.75            pl5321hec16e2b_3    bioconda\n",
    "perl-scope-guard          0.21            pl5321hdfd78af_3    bioconda\n",
    "perl-sub-info             0.002           pl5321hdfd78af_1    bioconda\n",
    "perl-term-table           0.016           pl5321hdfd78af_0    bioconda\n",
    "perl-test2-suite          0.000145        pl5321hdfd78af_0    bioconda\n",
    "perl-uri                  5.12            pl5321hdfd78af_0    bioconda\n",
    "perl-xml-libxml           2.0207          pl5321h661654b_0    bioconda\n",
    "perl-xml-namespacesupport 1.12            pl5321hdfd78af_1    bioconda\n",
    "perl-xml-sax              1.02            pl5321hdfd78af_1    bioconda\n",
    "perl-xml-sax-base         1.09            pl5321hdfd78af_1    bioconda\n",
    "pexpect                   4.9.0              pyhd8ed1ab_0    conda-forge\n",
    "picard                    3.2.0                hdfd78af_0    bioconda\n",
    "pickleshare               0.7.5                   py_1003    conda-forge\n",
    "pigz                      2.8                  h2797004_0    conda-forge\n",
    "pillow                    10.3.0          py312h5eee18b_0  \n",
    "pip                       24.0               pyhd8ed1ab_0    conda-forge\n",
    "pixman                    0.43.2               h59595ed_0    conda-forge\n",
    "pkgutil-resolve-name      1.3.10             pyhd8ed1ab_1    conda-forge\n",
    "plac                      1.4.3              pyhd8ed1ab_0    conda-forge\n",
    "platformdirs              4.2.2              pyhd8ed1ab_0    conda-forge\n",
    "pluggy                    1.5.0              pyhd8ed1ab_0    conda-forge\n",
    "prometheus_client         0.20.0             pyhd8ed1ab_0    conda-forge\n",
    "prompt-toolkit            3.0.47             pyha770c72_0    conda-forge\n",
    "prompt_toolkit            3.0.47               hd8ed1ab_0    conda-forge\n",
    "psutil                    6.0.0           py312h9a8786e_0    conda-forge\n",
    "pthread-stubs             0.4               h36c2ea0_1001    conda-forge\n",
    "ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge\n",
    "pulp                      2.8.0           py312h7900ff3_0    conda-forge\n",
    "pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge\n",
    "pycparser                 2.22               pyhd8ed1ab_0    conda-forge\n",
    "pydeseq2                  0.4.10             pyhdfd78af_0    bioconda\n",
    "pygithub                  2.3.0              pyhd8ed1ab_0    conda-forge\n",
    "pygments                  2.18.0             pyhd8ed1ab_0    conda-forge\n",
    "pyjwt                     2.8.0              pyhd8ed1ab_1    conda-forge\n",
    "pynacl                    1.5.0           py312h98912ed_3    conda-forge\n",
    "pyparsing                 3.1.2              pyhd8ed1ab_0    conda-forge\n",
    "pysocks                   1.7.1              pyha2e5f31_6    conda-forge\n",
    "pytest                    8.2.2              pyhd8ed1ab_0    conda-forge\n",
    "python                    3.12.3          hab00c5b_0_cpython    conda-forge\n",
    "python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge\n",
    "python-fastjsonschema     2.20.0             pyhd8ed1ab_0    conda-forge\n",
    "python-isal               1.6.1           py312h98912ed_0    conda-forge\n",
    "python-json-logger        2.0.7              pyhd8ed1ab_0    conda-forge\n",
    "python-tzdata             2024.1             pyhd8ed1ab_0    conda-forge\n",
    "python-zlib-ng            0.4.3           py312hdfe76eb_1    conda-forge\n",
    "python_abi                3.12                    4_cp312    conda-forge\n",
    "pytz                      2024.1             pyhd8ed1ab_0    conda-forge\n",
    "pyyaml                    6.0.1           py312h98912ed_1    conda-forge\n",
    "pyzmq                     26.0.3          py312h8fd38d8_0    conda-forge\n",
    "qhull                     2020.2               h4bd325d_2    conda-forge\n",
    "qtconsole-base            5.5.2              pyha770c72_0    conda-forge\n",
    "qtpy                      2.4.1              pyhd8ed1ab_0    conda-forge\n",
    "r-askpass                 1.2.0             r42h57805ef_0    conda-forge\n",
    "r-assertthat              0.2.1             r42hc72bb7e_4    conda-forge\n",
    "r-backports               1.5.0             r42hb1dbf0f_0    conda-forge\n",
    "r-base                    4.2.1                h7880091_2    conda-forge\n",
    "r-base64enc               0.1_3           r42h57805ef_1006    conda-forge\n",
    "r-bh                      1.84.0_0          r42hc72bb7e_0    conda-forge\n",
    "r-bit                     4.0.5             r42h57805ef_1    conda-forge\n",
    "r-bit64                   4.0.5             r42h57805ef_2    conda-forge\n",
    "r-bitops                  1.0_7             r42h57805ef_2    conda-forge\n",
    "r-blob                    1.2.4             r42hc72bb7e_1    conda-forge\n",
    "r-brew                    1.0_10            r42hc72bb7e_0    conda-forge\n",
    "r-broom                   1.0.6             r42hc72bb7e_0    conda-forge\n",
    "r-bslib                   0.7.0             r42hc72bb7e_0    conda-forge\n",
    "r-cachem                  1.1.0             r42hb1dbf0f_0    conda-forge\n",
    "r-callr                   3.7.6             r42hc72bb7e_0    conda-forge\n",
    "r-cellranger              1.1.0           r42hc72bb7e_1006    conda-forge\n",
    "r-cli                     3.6.3             r42ha18555a_0    conda-forge\n",
    "r-clipr                   0.8.0             r42hc72bb7e_2    conda-forge\n",
    "r-codetools               0.2_20            r42hc72bb7e_0    conda-forge\n",
    "r-collections             0.3.7             r42h57805ef_1    conda-forge\n",
    "r-colorspace              2.1_0             r42h57805ef_1    conda-forge\n",
    "r-commonmark              1.9.1             r42h57805ef_0    conda-forge\n",
    "r-cpp11                   0.4.7             r42hc72bb7e_0    conda-forge\n",
    "r-crayon                  1.5.3             r42hc72bb7e_0    conda-forge\n",
    "r-curl                    4.3.3             r42h06615bd_1    conda-forge\n",
    "r-cyclocomp               1.1.1             r42hc72bb7e_0    conda-forge\n",
    "r-data.table              1.15.2            r42h029312a_0    conda-forge\n",
    "r-dbi                     1.2.3             r42hc72bb7e_0    conda-forge\n",
    "r-dbplyr                  2.5.0             r42hc72bb7e_0    conda-forge\n",
    "r-desc                    1.4.3             r42hc72bb7e_0    conda-forge\n",
    "r-digest                  0.6.36            r42ha18555a_0    conda-forge\n",
    "r-dplyr                   1.1.4             r42ha503ecb_0    conda-forge\n",
    "r-dtplyr                  1.3.1             r42hc72bb7e_1    conda-forge\n",
    "r-ellipsis                0.3.2             r42h57805ef_2    conda-forge\n",
    "r-evaluate                0.24.0            r42hc72bb7e_0    conda-forge\n",
    "r-fansi                   1.0.6             r42h57805ef_0    conda-forge\n",
    "r-farver                  2.1.2             r42ha18555a_0    conda-forge\n",
    "r-fastmap                 1.2.0             r42ha18555a_0    conda-forge\n",
    "r-fontawesome             0.5.2             r42hc72bb7e_0    conda-forge\n",
    "r-forcats                 1.0.0             r42hc72bb7e_1    conda-forge\n",
    "r-formatr                 1.14              r42hc72bb7e_1    conda-forge\n",
    "r-fs                      1.6.4             r42ha18555a_0    conda-forge\n",
    "r-futile.logger           1.4.3           r42hc72bb7e_1005    conda-forge\n",
    "r-futile.options          1.0.1           r42hc72bb7e_1004    conda-forge\n",
    "r-gargle                  1.5.2             r42h785f33e_0    conda-forge\n",
    "r-generics                0.1.3             r42hc72bb7e_2    conda-forge\n",
    "r-ggplot2                 3.5.1             r42hc72bb7e_0    conda-forge\n",
    "r-glue                    1.7.0             r42h57805ef_0    conda-forge\n",
    "r-googledrive             2.1.1             r42hc72bb7e_1    conda-forge\n",
    "r-googlesheets4           1.1.1             r42h785f33e_1    conda-forge\n",
    "r-gtable                  0.3.5             r42hc72bb7e_0    conda-forge\n",
    "r-haven                   2.5.4             r42ha503ecb_0    conda-forge\n",
    "r-highr                   0.11              r42hc72bb7e_0    conda-forge\n",
    "r-hms                     1.1.3             r42hc72bb7e_1    conda-forge\n",
    "r-htmltools               0.5.8.1           r42ha503ecb_0    conda-forge\n",
    "r-httr                    1.4.7             r42hc72bb7e_0    conda-forge\n",
    "r-ids                     1.0.1             r42hc72bb7e_3    conda-forge\n",
    "r-isoband                 0.2.7             r42ha503ecb_2    conda-forge\n",
    "r-jquerylib               0.1.4             r42hc72bb7e_2    conda-forge\n",
    "r-jsonlite                1.8.8             r42h57805ef_0    conda-forge\n",
    "r-knitr                   1.47              r42hc72bb7e_0    conda-forge\n",
    "r-labeling                0.4.3             r42hc72bb7e_0    conda-forge\n",
    "r-lambda.r                1.2.4             r42hc72bb7e_3    conda-forge\n",
    "r-languageserver          0.3.16            r42h57805ef_0    conda-forge\n",
    "r-lattice                 0.22_6            r42h57805ef_0    conda-forge\n",
    "r-lazyeval                0.2.2             r42h57805ef_4    conda-forge\n",
    "r-lifecycle               1.0.4             r42hc72bb7e_0    conda-forge\n",
    "r-lintr                   3.1.2             r42hc72bb7e_0    conda-forge\n",
    "r-locfit                  1.5_9.9           r42h57805ef_0    conda-forge\n",
    "r-lubridate               1.9.3             r42h57805ef_0    conda-forge\n",
    "r-magrittr                2.0.3             r42h57805ef_2    conda-forge\n",
    "r-mass                    7.3_60.0.1        r42h57805ef_0    conda-forge\n",
    "r-matrix                  1.6_5             r42h316c678_0    conda-forge\n",
    "r-matrixstats             1.3.0             r42h57805ef_0    conda-forge\n",
    "r-memoise                 2.0.1             r42hc72bb7e_2    conda-forge\n",
    "r-mgcv                    1.9_1             r42h316c678_0    conda-forge\n",
    "r-mime                    0.12              r42h57805ef_2    conda-forge\n",
    "r-modelr                  0.1.11            r42hc72bb7e_1    conda-forge\n",
    "r-munsell                 0.5.1             r42hc72bb7e_0    conda-forge\n",
    "r-nlme                    3.1_165           r42hbcb9c34_0    conda-forge\n",
    "r-openssl                 2.2.0             r42hd7d0243_0    conda-forge\n",
    "r-pillar                  1.9.0             r42hc72bb7e_1    conda-forge\n",
    "r-pkgbuild                1.4.4             r42hc72bb7e_0    conda-forge\n",
    "r-pkgconfig               2.0.3             r42hc72bb7e_3    conda-forge\n",
    "r-pkgload                 1.3.4             r42hc72bb7e_0    conda-forge\n",
    "r-plogr                   0.2.0           r42hc72bb7e_1005    conda-forge\n",
    "r-png                     0.1_8             r42h81d01c5_1    conda-forge\n",
    "r-prettyunits             1.2.0             r42hc72bb7e_0    conda-forge\n",
    "r-processx                3.8.4             r42h57805ef_0    conda-forge\n",
    "r-progress                1.2.3             r42hc72bb7e_0    conda-forge\n",
    "r-ps                      1.7.6             r42h57805ef_0    conda-forge\n",
    "r-purrr                   1.0.2             r42h57805ef_0    conda-forge\n",
    "r-r.cache                 0.16.0            r42hc72bb7e_2    conda-forge\n",
    "r-r.methodss3             1.8.2             r42hc72bb7e_2    conda-forge\n",
    "r-r.oo                    1.26.0            r42hc72bb7e_0    conda-forge\n",
    "r-r.utils                 2.12.3            r42hc72bb7e_0    conda-forge\n",
    "r-r6                      2.5.1             r42hc72bb7e_2    conda-forge\n",
    "r-rappdirs                0.3.3             r42h57805ef_2    conda-forge\n",
    "r-rcolorbrewer            1.1_3             r42h785f33e_2    conda-forge\n",
    "r-rcpp                    1.0.12            r42h7df8631_0    conda-forge\n",
    "r-rcpparmadillo           0.12.8.4.0        r42h58a4165_0    conda-forge\n",
    "r-rcurl                   1.98_1.9          r42h06615bd_1    conda-forge\n",
    "r-readr                   2.1.5             r42ha503ecb_0    conda-forge\n",
    "r-readxl                  1.4.3             r42ha5c9fba_0    conda-forge\n",
    "r-rematch                 2.0.0             r42hc72bb7e_0    conda-forge\n",
    "r-rematch2                2.1.2             r42hc72bb7e_3    conda-forge\n",
    "r-remotes                 2.5.0             r42hc72bb7e_0    conda-forge\n",
    "r-repr                    1.1.7             r42h785f33e_0    conda-forge\n",
    "r-reprex                  2.1.0             r42hc72bb7e_0    conda-forge\n",
    "r-rex                     1.2.1             r42hc72bb7e_2    conda-forge\n",
    "r-rlang                   1.1.4             r42ha18555a_0    conda-forge\n",
    "r-rmarkdown               2.27              r42hc72bb7e_1    conda-forge\n",
    "r-roxygen2                7.3.2             r42ha18555a_0    conda-forge\n",
    "r-rprojroot               2.0.4             r42hc72bb7e_0    conda-forge\n",
    "r-rsqlite                 2.3.4             r42ha503ecb_0    conda-forge\n",
    "r-rstudioapi              0.16.0            r42hc72bb7e_0    conda-forge\n",
    "r-rvest                   1.0.4             r42hc72bb7e_0    conda-forge\n",
    "r-sass                    0.4.9             r42ha503ecb_0    conda-forge\n",
    "r-scales                  1.3.0             r42hc72bb7e_0    conda-forge\n",
    "r-selectr                 0.4_2             r42hc72bb7e_3    conda-forge\n",
    "r-snow                    0.4_4             r42hc72bb7e_2    conda-forge\n",
    "r-stringi                 1.7.12            r42h1ae9187_0    conda-forge\n",
    "r-stringr                 1.5.1             r42h785f33e_0    conda-forge\n",
    "r-styler                  1.10.3            r42hc72bb7e_0    conda-forge\n",
    "r-survival                3.7_0             r42hb1dbf0f_0    conda-forge\n",
    "r-sys                     3.4.2             r42h57805ef_1    conda-forge\n",
    "r-tibble                  3.2.1             r42h57805ef_2    conda-forge\n",
    "r-tidyr                   1.3.1             r42ha503ecb_0    conda-forge\n",
    "r-tidyselect              1.2.1             r42hc72bb7e_0    conda-forge\n",
    "r-tidyverse               1.3.2             r42hc72bb7e_1    conda-forge\n",
    "r-timechange              0.3.0             r42ha503ecb_0    conda-forge\n",
    "r-tinytex                 0.51              r42hc72bb7e_0    conda-forge\n",
    "r-tzdb                    0.4.0             r42ha503ecb_1    conda-forge\n",
    "r-utf8                    1.2.4             r42h57805ef_0    conda-forge\n",
    "r-uuid                    1.2_0             r42h57805ef_0    conda-forge\n",
    "r-vctrs                   0.6.5             r42ha503ecb_0    conda-forge\n",
    "r-viridislite             0.4.2             r42hc72bb7e_1    conda-forge\n",
    "r-vroom                   1.6.5             r42ha503ecb_0    conda-forge\n",
    "r-withr                   3.0.0             r42hc72bb7e_0    conda-forge\n",
    "r-xfun                    0.45              r42ha18555a_0    conda-forge\n",
    "r-xml                     3.99_0.11         r42h2b86b34_2    conda-forge\n",
    "r-xml2                    1.3.3             r42h044e5c7_2    conda-forge\n",
    "r-xmlparsedata            1.0.5             r42hc72bb7e_2    conda-forge\n",
    "r-xtable                  1.8_4             r42hc72bb7e_5    conda-forge\n",
    "r-yaml                    2.3.8             r42h57805ef_0    conda-forge\n",
    "readline                  8.2                  h8228510_1    conda-forge\n",
    "referencing               0.35.1             pyhd8ed1ab_0    conda-forge\n",
    "requests                  2.32.3             pyhd8ed1ab_0    conda-forge\n",
    "reretry                   0.11.8             pyhd8ed1ab_0    conda-forge\n",
    "rfc3339-validator         0.1.4              pyhd8ed1ab_0    conda-forge\n",
    "rfc3986-validator         0.1.1              pyh9f0ad1d_0    conda-forge\n",
    "rich                      13.7.1             pyhd8ed1ab_0    conda-forge\n",
    "rich-click                1.8.3              pyhd8ed1ab_0    conda-forge\n",
    "rpds-py                   0.18.1          py312h4413252_0    conda-forge\n",
    "samtools                  1.6                 hc3601fc_10    bioconda\n",
    "scikit-learn              1.5.1           py312h775a589_0    conda-forge\n",
    "scipy                     1.14.0          py312hc2bc53b_0    conda-forge\n",
    "sed                       4.8                  he412f7d_0    conda-forge\n",
    "send2trash                1.8.3              pyh0d859eb_0    conda-forge\n",
    "setuptools                70.1.1             pyhd8ed1ab_0    conda-forge\n",
    "simplejson                3.19.2          py312h98912ed_0    conda-forge\n",
    "six                       1.16.0             pyh6c4a22f_0    conda-forge\n",
    "slack-sdk                 3.30.0             pyha770c72_0    conda-forge\n",
    "slack_sdk                 3.30.0               hd8ed1ab_0    conda-forge\n",
    "smart_open                7.0.4              pyhd8ed1ab_0    conda-forge\n",
    "smmap                     5.0.0              pyhd8ed1ab_0    conda-forge\n",
    "snakedeploy               0.10.0             pyhdfd78af_0    bioconda\n",
    "snakemake                 8.14.0               hdfd78af_0    bioconda\n",
    "snakemake-interface-common 1.17.2             pyhdfd78af_0    bioconda\n",
    "snakemake-interface-executor-plugins 9.1.1              pyhdfd78af_0    bioconda\n",
    "snakemake-interface-report-plugins 1.0.0              pyhdfd78af_0    bioconda\n",
    "snakemake-interface-storage-plugins 3.2.2              pyhdfd78af_0    bioconda\n",
    "snakemake-minimal         8.14.0             pyhdfd78af_0    bioconda\n",
    "sniffio                   1.3.1              pyhd8ed1ab_0    conda-forge\n",
    "soupsieve                 2.5                pyhd8ed1ab_1    conda-forge\n",
    "spectra                   0.0.11                     py_1    conda-forge\n",
    "sra-tools                 3.1.1                h4304569_0    bioconda\n",
    "stack_data                0.6.2              pyhd8ed1ab_0    conda-forge\n",
    "star                      2.7.10b              h9ee0642_0    bioconda\n",
    "stopit                    1.1.2                      py_0    conda-forge\n",
    "subread                   2.0.6                he4a0461_2    bioconda\n",
    "sysroot_linux-64          2.12                he073ed8_17    conda-forge\n",
    "tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge\n",
    "terminado                 0.18.1             pyh0d859eb_0    conda-forge\n",
    "threadpoolctl             3.5.0              pyhc1e730c_0    conda-forge\n",
    "throttler                 1.2.2              pyhd8ed1ab_0    conda-forge\n",
    "tinycss2                  1.3.0              pyhd8ed1ab_0    conda-forge\n",
    "tk                        8.6.13          noxft_h4845f30_101    conda-forge\n",
    "tktable                   2.10                 h8bc8fbc_6    conda-forge\n",
    "toml                      0.10.2             pyhd8ed1ab_0    conda-forge\n",
    "tomli                     2.0.1              pyhd8ed1ab_0    conda-forge\n",
    "tomlkit                   0.13.2             pyha770c72_0    conda-forge\n",
    "toposort                  1.10               pyhd8ed1ab_0    conda-forge\n",
    "tornado                   6.4.1           py312h9a8786e_0    conda-forge\n",
    "traitlets                 5.14.3             pyhd8ed1ab_0    conda-forge\n",
    "types-python-dateutil     2.9.0.20240316     pyhd8ed1ab_0    conda-forge\n",
    "typing-extensions         4.12.2               hd8ed1ab_0    conda-forge\n",
    "typing_extensions         4.12.2             pyha770c72_0    conda-forge\n",
    "typing_utils              0.1.0              pyhd8ed1ab_0    conda-forge\n",
    "tzdata                    2024a                h0c530f3_0    conda-forge\n",
    "ubiquerg                  0.8.0              pyhd8ed1ab_0    conda-forge\n",
    "uri-template              1.3.0              pyhd8ed1ab_0    conda-forge\n",
    "urllib3                   2.2.2              pyhd8ed1ab_1    conda-forge\n",
    "veracitools               0.1.3                      py_0    conda-forge\n",
    "wcwidth                   0.2.13             pyhd8ed1ab_0    conda-forge\n",
    "webcolors                 24.6.0             pyhd8ed1ab_0    conda-forge\n",
    "webencodings              0.5.1              pyhd8ed1ab_2    conda-forge\n",
    "websocket-client          1.8.0              pyhd8ed1ab_0    conda-forge\n",
    "wheel                     0.43.0             pyhd8ed1ab_1    conda-forge\n",
    "widgetsnbextension        4.0.11             pyhd8ed1ab_0    conda-forge\n",
    "wrapt                     1.16.0          py312h98912ed_0    conda-forge\n",
    "xmltodict                 0.14.2             pyhd8ed1ab_0    conda-forge\n",
    "xopen                     2.0.1           py312h7900ff3_0    conda-forge\n",
    "xorg-fixesproto           5.0               h7f98852_1002    conda-forge\n",
    "xorg-inputproto           2.3.2             h7f98852_1002    conda-forge\n",
    "xorg-kbproto              1.0.7             h7f98852_1002    conda-forge\n",
    "xorg-libice               1.1.1                hd590300_0    conda-forge\n",
    "xorg-libsm                1.2.4                h7391055_0    conda-forge\n",
    "xorg-libx11               1.8.9                h8ee46fc_0    conda-forge\n",
    "xorg-libxau               1.0.11               hd590300_0    conda-forge\n",
    "xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge\n",
    "xorg-libxext              1.3.4                h0b41bf4_2    conda-forge\n",
    "xorg-libxfixes            5.0.3             h7f98852_1004    conda-forge\n",
    "xorg-libxi                1.7.10               h7f98852_0    conda-forge\n",
    "xorg-libxrender           0.9.11               hd590300_0    conda-forge\n",
    "xorg-libxt                1.3.0                hd590300_1    conda-forge\n",
    "xorg-libxtst              1.2.3             h7f98852_1002    conda-forge\n",
    "xorg-recordproto          1.14.2            h7f98852_1002    conda-forge\n",
    "xorg-renderproto          0.11.1            h7f98852_1002    conda-forge\n",
    "xorg-xextproto            7.3.0             h0b41bf4_1003    conda-forge\n",
    "xorg-xproto               7.0.31            h7f98852_1007    conda-forge\n",
    "xz                        5.2.6                h166bdaf_0    conda-forge\n",
    "yaml                      0.2.5                h7f98852_2    conda-forge\n",
    "yq                        3.4.3              pyhd8ed1ab_0    conda-forge\n",
    "yte                       1.5.4              pyha770c72_0    conda-forge\n",
    "zeromq                    4.3.5                h59595ed_1    conda-forge\n",
    "zipp                      3.19.2             pyhd8ed1ab_0    conda-forge\n",
    "zlib                      1.2.13               h4ab18f5_6    conda-forge\n",
    "zlib-ng                   2.2.1                he02047a_0    conda-forge\n",
    "zstandard                 0.22.0          py312h5b18bf6_1    conda-forge\n",
    "zstd                      1.5.6                ha6fb4c9_0    conda-forge\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#\n",
    "# Name                    Version                   Build  Channel\n",
    "_libgcc_mutex             0.1                 conda_forge    conda-forge\n",
    "_openmp_mutex             4.5                       2_gnu    conda-forge\n",
    "_r-mutex                  1.0.1               anacondar_1    conda-forge\n",
    "alsa-lib                  1.2.13               hb9d3cd8_0    conda-forge\n",
    "amply                     0.1.6              pyhd8ed1ab_0    conda-forge\n",
    "annotated-types           0.7.0              pyhd8ed1ab_1    conda-forge\n",
    "appdirs                   1.4.4              pyhd8ed1ab_1    conda-forge\n",
    "argcomplete               3.5.2              pyhd8ed1ab_0    conda-forge\n",
    "argparse-dataclass        2.0.0              pyhd8ed1ab_0    conda-forge\n",
    "asttokens                 3.0.0              pyhd8ed1ab_1    conda-forge\n",
    "attrs                     24.2.0             pyh71513ae_1    conda-forge\n",
    "binutils_impl_linux-64    2.43                 h4bf12b8_2    conda-forge\n",
    "bioconductor-apeglm       1.24.0            r43hf17093f_1    bioconda\n",
    "bioconductor-biobase      2.62.0            r43ha9d7317_3    bioconda\n",
    "bioconductor-biocgenerics 0.48.1            r43hdfd78af_2    bioconda\n",
    "bioconductor-biocio       1.12.0            r43hdfd78af_0    bioconda\n",
    "bioconductor-biocparallel 1.36.0            r43hf17093f_2    bioconda\n",
    "bioconductor-biostrings   2.70.1            r43ha9d7317_2    bioconda\n",
    "bioconductor-data-packages 20231203             hdfd78af_0    bioconda\n",
    "bioconductor-delayedarray 0.28.0            r43ha9d7317_2    bioconda\n",
    "bioconductor-deseq2       1.42.0            r43hf17093f_2    bioconda\n",
    "bioconductor-genomeinfodb 1.38.1            r43hdfd78af_1    bioconda\n",
    "bioconductor-genomeinfodbdata 1.2.11            r43hdfd78af_1    bioconda\n",
    "bioconductor-genomicalignments 1.38.0            r43ha9d7317_1    bioconda\n",
    "bioconductor-genomicranges 1.54.1            r43ha9d7317_2    bioconda\n",
    "bioconductor-iranges      2.36.0            r43ha9d7317_2    bioconda\n",
    "bioconductor-matrixgenerics 1.14.0            r43hdfd78af_3    bioconda\n",
    "bioconductor-noiseq       2.46.0            r43hdfd78af_0    bioconda\n",
    "bioconductor-rhtslib      2.4.0             r43ha9d7317_2    bioconda\n",
    "bioconductor-rsamtools    2.18.0            r43hf17093f_2    bioconda\n",
    "bioconductor-rtracklayer  1.62.0            r43ha9d7317_1    bioconda\n",
    "bioconductor-s4arrays     1.2.0             r43ha9d7317_2    bioconda\n",
    "bioconductor-s4vectors    0.40.2            r43ha9d7317_2    bioconda\n",
    "bioconductor-sparsearray  1.2.2             r43ha9d7317_2    bioconda\n",
    "bioconductor-summarizedexperiment 1.32.0            r43hdfd78af_0    bioconda\n",
    "bioconductor-xvector      0.42.0            r43ha9d7317_2    bioconda\n",
    "bioconductor-zlibbioc     1.48.0            r43ha9d7317_2    bioconda\n",
    "brotli-python             1.1.0           py312h2ec8cdc_2    conda-forge\n",
    "bwidget                   1.10.1               ha770c72_0    conda-forge\n",
    "bzip2                     1.0.8                h4bc722e_7    conda-forge\n",
    "c-ares                    1.34.3               hb9d3cd8_1    conda-forge\n",
    "ca-certificates           2024.8.30            hbcca054_0    conda-forge\n",
    "cairo                     1.18.2               h3394656_1    conda-forge\n",
    "certifi                   2024.8.30          pyhd8ed1ab_0    conda-forge\n",
    "cffi                      1.17.1          py312h06ac9bb_0    conda-forge\n",
    "charset-normalizer        3.4.0              pyhd8ed1ab_1    conda-forge\n",
    "click                     8.1.7           unix_pyh707e725_1    conda-forge\n",
    "coin-or-cbc               2.10.12              h8b142ea_1    conda-forge\n",
    "coin-or-cgl               0.60.9               h1d3f3f2_0    conda-forge\n",
    "coin-or-clp               1.17.10              h07f2a63_0    conda-forge\n",
    "coin-or-osi               0.108.11             h6514dde_1    conda-forge\n",
    "coin-or-utils             2.11.12              h99da652_1    conda-forge\n",
    "coincbc                   2.10.12           1_metapackage    conda-forge\n",
    "colorama                  0.4.6              pyhd8ed1ab_1    conda-forge\n",
    "coloredlogs               15.0.1             pyhd8ed1ab_4    conda-forge\n",
    "colormath                 3.0.0                      py_2    conda-forge\n",
    "comm                      0.2.2              pyhd8ed1ab_1    conda-forge\n",
    "conda-inject              1.3.2              pyhd8ed1ab_0    conda-forge\n",
    "configargparse            1.7                pyhd8ed1ab_0    conda-forge\n",
    "connection_pool           0.0.3              pyhd3deb0d_0    conda-forge\n",
    "curl                      8.11.1               h332b0f4_0    conda-forge\n",
    "cutadapt                  5.0             py312h0fa9677_0    bioconda\n",
    "datrie                    0.8.2           py312h66e93f0_8    conda-forge\n",
    "debugpy                   1.8.11          py312h2ec8cdc_0    conda-forge\n",
    "decorator                 5.1.1              pyhd8ed1ab_1    conda-forge\n",
    "dnaio                     1.2.2           py312hf67a6ed_0    bioconda\n",
    "docutils                  0.21.2             pyhd8ed1ab_1    conda-forge\n",
    "dpath                     2.2.0              pyha770c72_0    conda-forge\n",
    "eido                      0.2.4              pyhd8ed1ab_0    conda-forge\n",
    "exceptiongroup            1.2.2              pyhd8ed1ab_1    conda-forge\n",
    "executing                 2.1.0              pyhd8ed1ab_1    conda-forge\n",
    "expat                     2.6.4                h5888daf_0    conda-forge\n",
    "fastqc                    0.12.1               hdfd78af_0    bioconda\n",
    "font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge\n",
    "font-ttf-inconsolata      3.000                h77eed37_0    conda-forge\n",
    "font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge\n",
    "font-ttf-ubuntu           0.83                 h77eed37_3    conda-forge\n",
    "fontconfig                2.15.0               h7e30c49_1    conda-forge\n",
    "fonts-conda-ecosystem     1                             0    conda-forge\n",
    "fonts-conda-forge         1                             0    conda-forge\n",
    "freetype                  2.12.1               h267a509_2    conda-forge\n",
    "fribidi                   1.0.10               h36c2ea0_0    conda-forge\n",
    "gcc_impl_linux-64         14.2.0               h6b349bd_1    conda-forge\n",
    "gfortran_impl_linux-64    14.2.0               hc73f493_1    conda-forge\n",
    "giflib                    5.2.2                hd590300_0    conda-forge\n",
    "gitdb                     4.0.11             pyhd8ed1ab_1    conda-forge\n",
    "gitpython                 3.1.43             pyhff2d567_1    conda-forge\n",
    "graphite2                 1.3.13            h59595ed_1003    conda-forge\n",
    "gsl                       2.7                  he838d99_0    conda-forge\n",
    "gxx_impl_linux-64         14.2.0               h2c03514_1    conda-forge\n",
    "h2                        4.1.0              pyhd8ed1ab_1    conda-forge\n",
    "harfbuzz                  9.0.0                hda332d3_1    conda-forge\n",
    "hisat2                    2.2.1                h503566f_8    bioconda\n",
    "hpack                     4.0.0              pyhd8ed1ab_1    conda-forge\n",
    "htslib                    1.21                 h566b1c6_1    bioconda\n",
    "humanfriendly             10.0               pyh707e725_8    conda-forge\n",
    "humanize                  4.11.0             pyhd8ed1ab_1    conda-forge\n",
    "hyperframe                6.0.1              pyhd8ed1ab_1    conda-forge\n",
    "icu                       75.1                 he02047a_0    conda-forge\n",
    "idna                      3.10               pyhd8ed1ab_1    conda-forge\n",
    "immutables                0.21            py312h66e93f0_0    conda-forge\n",
    "importlib-metadata        8.5.0              pyha770c72_1    conda-forge\n",
    "importlib_resources       6.4.5              pyhd8ed1ab_1    conda-forge\n",
    "iniconfig                 2.0.0              pyhd8ed1ab_1    conda-forge\n",
    "ipykernel                 6.29.5             pyh3099207_0    conda-forge\n",
    "ipython                   8.30.0             pyh707e725_0    conda-forge\n",
    "isa-l                     2.31.0               h4bc722e_2    conda-forge\n",
    "jedi                      0.19.2             pyhd8ed1ab_1    conda-forge\n",
    "jinja2                    3.1.4              pyhd8ed1ab_1    conda-forge\n",
    "jq                        1.7.1                hd590300_0    conda-forge\n",
    "jsonschema                4.23.0             pyhd8ed1ab_1    conda-forge\n",
    "jsonschema-specifications 2024.10.1          pyhd8ed1ab_1    conda-forge\n",
    "jupyter_client            8.6.3              pyhd8ed1ab_1    conda-forge\n",
    "jupyter_core              5.7.2              pyh31011fe_1    conda-forge\n",
    "kaleido-core              0.2.1                h3644ca4_0    conda-forge\n",
    "kernel-headers_linux-64   3.10.0              he073ed8_18    conda-forge\n",
    "keyutils                  1.6.1                h166bdaf_0    conda-forge\n",
    "krb5                      1.21.3               h659f571_0    conda-forge\n",
    "lcms2                     2.16                 hb7c19ff_0    conda-forge\n",
    "ld_impl_linux-64          2.43                 h712a8e2_2    conda-forge\n",
    "lerc                      4.0.0                h27087fc_0    conda-forge\n",
    "libblas                   3.9.0           25_linux64_openblas    conda-forge\n",
    "libcblas                  3.9.0           25_linux64_openblas    conda-forge\n",
    "libcups                   2.3.3                h4637d8d_4    conda-forge\n",
    "libcurl                   8.11.1               h332b0f4_0    conda-forge\n",
    "libdeflate                1.22                 hb9d3cd8_0    conda-forge\n",
    "libedit                   3.1.20191231         he28a2e2_2    conda-forge\n",
    "libev                     4.33                 hd590300_2    conda-forge\n",
    "libexpat                  2.6.4                h5888daf_0    conda-forge\n",
    "libffi                    3.4.2                h7f98852_5    conda-forge\n",
    "libgcc                    14.2.0               h77fa898_1    conda-forge\n",
    "libgcc-devel_linux-64     14.2.0             h41c2201_101    conda-forge\n",
    "libgcc-ng                 14.2.0               h69a702a_1    conda-forge\n",
    "libgfortran               14.2.0               h69a702a_1    conda-forge\n",
    "libgfortran-ng            14.2.0               h69a702a_1    conda-forge\n",
    "libgfortran5              14.2.0               hd5240d6_1    conda-forge\n",
    "libglib                   2.82.2               h2ff4ddf_0    conda-forge\n",
    "libgomp                   14.2.0               h77fa898_1    conda-forge\n",
    "libiconv                  1.17                 hd590300_2    conda-forge\n",
    "libjpeg-turbo             3.0.0                hd590300_1    conda-forge\n",
    "liblapack                 3.9.0           25_linux64_openblas    conda-forge\n",
    "liblapacke                3.9.0           25_linux64_openblas    conda-forge\n",
    "liblzma                   5.6.3                hb9d3cd8_1    conda-forge\n",
    "liblzma-devel             5.6.3                hb9d3cd8_1    conda-forge\n",
    "libnghttp2                1.64.0               h161d5f1_0    conda-forge\n",
    "libnsl                    2.0.1                hd590300_0    conda-forge\n",
    "libopenblas               0.3.28          pthreads_h94d23a6_1    conda-forge\n",
    "libpng                    1.6.44               hadc24fc_0    conda-forge\n",
    "libsanitizer              14.2.0               h2a3dede_1    conda-forge\n",
    "libsodium                 1.0.20               h4ab18f5_0    conda-forge\n",
    "libsqlite                 3.47.2               hee588c1_0    conda-forge\n",
    "libssh2                   1.11.1               hf672d98_0    conda-forge\n",
    "libstdcxx                 14.2.0               hc0a3c3a_1    conda-forge\n",
    "libstdcxx-devel_linux-64  14.2.0             h41c2201_101    conda-forge\n",
    "libstdcxx-ng              14.2.0               h4852527_1    conda-forge\n",
    "libtiff                   4.7.0                hc4654cb_2    conda-forge\n",
    "libuuid                   2.38.1               h0b41bf4_0    conda-forge\n",
    "libwebp-base              1.4.0                hd590300_0    conda-forge\n",
    "libxcb                    1.17.0               h8a09558_0    conda-forge\n",
    "libxcrypt                 4.4.36               hd590300_1    conda-forge\n",
    "libxml2                   2.13.5               h8d12d68_1    conda-forge\n",
    "libzlib                   1.3.1                hb9d3cd8_2    conda-forge\n",
    "logmuse                   0.2.8              pyhd8ed1ab_0    conda-forge\n",
    "make                      4.4.1                hb9d3cd8_2    conda-forge\n",
    "markdown                  3.6                pyhd8ed1ab_0    conda-forge\n",
    "markdown-it-py            3.0.0              pyhd8ed1ab_1    conda-forge\n",
    "markupsafe                3.0.2           py312h178313f_1    conda-forge\n",
    "mathjax                   2.7.7                ha770c72_3    conda-forge\n",
    "matplotlib-inline         0.1.7              pyhd8ed1ab_1    conda-forge\n",
    "mdurl                     0.1.2              pyhd8ed1ab_1    conda-forge\n",
    "multiqc                   1.25.2             pyhdfd78af_0    bioconda\n",
    "natsort                   8.4.0              pyh29332c3_1    conda-forge\n",
    "nbformat                  5.10.4             pyhd8ed1ab_1    conda-forge\n",
    "ncurses                   6.5                  he02047a_1    conda-forge\n",
    "nest-asyncio              1.6.0              pyhd8ed1ab_1    conda-forge\n",
    "networkx                  3.4.2              pyh267e887_2    conda-forge\n",
    "nspr                      4.36                 h5888daf_0    conda-forge\n",
    "nss                       3.107                hdf54f9c_0    conda-forge\n",
    "numpy                     2.2.0           py312h7e784f5_0    conda-forge\n",
    "oniguruma                 6.9.9                hd590300_0    conda-forge\n",
    "openjdk                   23.0.1               h4c11d01_0    conda-forge\n",
    "openjpeg                  2.5.3                h5fbd93e_0    conda-forge\n",
    "openssl                   3.4.0                hb9d3cd8_0    conda-forge\n",
    "packaging                 24.2               pyhd8ed1ab_2    conda-forge\n",
    "pandas                    2.2.2           py312h1d6d2e6_1    conda-forge\n",
    "pandoc                    3.6                  ha770c72_0    conda-forge\n",
    "pango                     1.54.0               h3a902e7_3    conda-forge\n",
    "parso                     0.8.4              pyhd8ed1ab_1    conda-forge\n",
    "pbzip2                    1.1.13               h1fcc475_2    conda-forge\n",
    "pcre2                     10.44                hba22ea6_2    conda-forge\n",
    "pephubclient              0.4.4              pyhd8ed1ab_0    conda-forge\n",
    "peppy                     0.40.7             pyhd8ed1ab_1    conda-forge\n",
    "perl                      5.32.1          7_hd590300_perl5    conda-forge\n",
    "pexpect                   4.9.0              pyhd8ed1ab_1    conda-forge\n",
    "picard                    3.3.0                hdfd78af_0    bioconda\n",
    "pickleshare               0.7.5           pyhd8ed1ab_1004    conda-forge\n",
    "pigz                      2.8                  h2797004_0    conda-forge\n",
    "pillow                    11.0.0          py312h7b63e92_0    conda-forge\n",
    "pip                       24.3.1             pyh8b19718_0    conda-forge\n",
    "pixman                    0.44.2               h29eaf8c_0    conda-forge\n",
    "pkgutil-resolve-name      1.3.10             pyhd8ed1ab_2    conda-forge\n",
    "plac                      1.4.3              pyhd8ed1ab_0    conda-forge\n",
    "platformdirs              4.3.6              pyhd8ed1ab_1    conda-forge\n",
    "plotly                    5.24.1             pyhd8ed1ab_1    conda-forge\n",
    "pluggy                    1.5.0              pyhd8ed1ab_1    conda-forge\n",
    "prompt-toolkit            3.0.48             pyha770c72_1    conda-forge\n",
    "psutil                    6.1.0           py312h66e93f0_0    conda-forge\n",
    "pthread-stubs             0.4               hb9d3cd8_1002    conda-forge\n",
    "ptyprocess                0.7.0              pyhd8ed1ab_1    conda-forge\n",
    "pulp                      2.8.0           py312h7900ff3_0    conda-forge\n",
    "pure_eval                 0.2.3              pyhd8ed1ab_1    conda-forge\n",
    "pyaml-env                 1.2.1              pyhd8ed1ab_0    conda-forge\n",
    "pycparser                 2.22               pyh29332c3_1    conda-forge\n",
    "pydantic                  2.10.3             pyh3cfb1c2_0    conda-forge\n",
    "pydantic-core             2.27.1          py312h12e396e_0    conda-forge\n",
    "pygments                  2.18.0             pyhd8ed1ab_1    conda-forge\n",
    "pyparsing                 3.2.0              pyhd8ed1ab_2    conda-forge\n",
    "pysocks                   1.7.1              pyha55dd90_7    conda-forge\n",
    "pytest                    8.3.4              pyhd8ed1ab_1    conda-forge\n",
    "python                    3.12.8          h9e4cc4f_1_cpython    conda-forge\n",
    "python-dateutil           2.9.0.post0        pyhff2d567_1    conda-forge\n",
    "python-fastjsonschema     2.21.1             pyhd8ed1ab_0    conda-forge\n",
    "python-isal               1.7.1           py312h66e93f0_0    conda-forge\n",
    "python-kaleido            0.2.1              pyhd8ed1ab_0    conda-forge\n",
    "python-tzdata             2024.2             pyhd8ed1ab_1    conda-forge\n",
    "python-zlib-ng            0.5.1           py312h27ca174_0    conda-forge\n",
    "python_abi                3.12                    5_cp312    conda-forge\n",
    "pytz                      2024.2             pyhd8ed1ab_1    conda-forge\n",
    "pyyaml                    6.0.2           py312h66e93f0_1    conda-forge\n",
    "pyzmq                     26.2.0          py312hbf22597_3    conda-forge\n",
    "qualimap                  2.3                  hdfd78af_0    bioconda\n",
    "r-abind                   1.4_5           r43hc72bb7e_1006    conda-forge\n",
    "r-ashr                    2.2_63            r43h0d4f4ea_1    conda-forge\n",
    "r-askpass                 1.2.1             r43h2b5f3a1_0    conda-forge\n",
    "r-assertthat              0.2.1             r43hc72bb7e_5    conda-forge\n",
    "r-backports               1.5.0             r43hb1dbf0f_1    conda-forge\n",
    "r-base                    4.3.3               h68df6d5_15    conda-forge\n",
    "r-base64enc               0.1_3           r43hb1dbf0f_1007    conda-forge\n",
    "r-bbmle                   1.0.25.1          r43hc72bb7e_1    conda-forge\n",
    "r-bdsmatrix               1.3_7             r43hb1dbf0f_1    conda-forge\n",
    "r-bh                      1.84.0_0          r43hc72bb7e_1    conda-forge\n",
    "r-bit                     4.5.0.1           r43h2b5f3a1_0    conda-forge\n",
    "r-bit64                   4.5.2             r43h2b5f3a1_0    conda-forge\n",
    "r-bitops                  1.0_9             r43h2b5f3a1_0    conda-forge\n",
    "r-blob                    1.2.4             r43hc72bb7e_2    conda-forge\n",
    "r-brew                    1.0_10            r43hc72bb7e_1    conda-forge\n",
    "r-broom                   1.0.7             r43hc72bb7e_0    conda-forge\n",
    "r-bslib                   0.8.0             r43hc72bb7e_0    conda-forge\n",
    "r-cachem                  1.1.0             r43hb1dbf0f_1    conda-forge\n",
    "r-callr                   3.7.6             r43hc72bb7e_1    conda-forge\n",
    "r-cellranger              1.1.0           r43hc72bb7e_1007    conda-forge\n",
    "r-cli                     3.6.3             r43h0d4f4ea_1    conda-forge\n",
    "r-clipr                   0.8.0             r43hc72bb7e_3    conda-forge\n",
    "r-coda                    0.19_4.1          r43hc72bb7e_1    conda-forge\n",
    "r-codetools               0.2_20            r43hc72bb7e_1    conda-forge\n",
    "r-collections             0.3.7             r43hb1dbf0f_2    conda-forge\n",
    "r-colorspace              2.1_1             r43hdb488b9_0    conda-forge\n",
    "r-commonmark              1.9.2             r43h2b5f3a1_0    conda-forge\n",
    "r-conflicted              1.2.0             r43h785f33e_2    conda-forge\n",
    "r-cpp11                   0.5.1             r43hc72bb7e_0    conda-forge\n",
    "r-crayon                  1.5.3             r43hc72bb7e_1    conda-forge\n",
    "r-curl                    6.0.1             r43h2700575_0    conda-forge\n",
    "r-cyclocomp               1.1.1             r43hc72bb7e_1    conda-forge\n",
    "r-data.table              1.15.4            r43h5f06984_1    conda-forge\n",
    "r-dbi                     1.2.3             r43hc72bb7e_1    conda-forge\n",
    "r-dbplyr                  2.5.0             r43hc72bb7e_1    conda-forge\n",
    "r-desc                    1.4.3             r43hc72bb7e_1    conda-forge\n",
    "r-digest                  0.6.37            r43h0d4f4ea_0    conda-forge\n",
    "r-dplyr                   1.1.4             r43h0d4f4ea_1    conda-forge\n",
    "r-dtplyr                  1.3.1             r43hc72bb7e_2    conda-forge\n",
    "r-ellipsis                0.3.2             r43hb1dbf0f_3    conda-forge\n",
    "r-emdbook                 1.3.13            r43hc72bb7e_1    conda-forge\n",
    "r-etrunct                 0.1             r43hc72bb7e_1006    conda-forge\n",
    "r-evaluate                1.0.1             r43hc72bb7e_0    conda-forge\n",
    "r-fansi                   1.0.6             r43hb1dbf0f_1    conda-forge\n",
    "r-farver                  2.1.2             r43ha18555a_1    conda-forge\n",
    "r-fastmap                 1.2.0             r43ha18555a_1    conda-forge\n",
    "r-fontawesome             0.5.3             r43hc72bb7e_0    conda-forge\n",
    "r-forcats                 1.0.0             r43hc72bb7e_2    conda-forge\n",
    "r-formatr                 1.14              r43hc72bb7e_2    conda-forge\n",
    "r-fs                      1.6.5             r43h93ab643_0    conda-forge\n",
    "r-futile.logger           1.4.3           r43hc72bb7e_1006    conda-forge\n",
    "r-futile.options          1.0.1           r43hc72bb7e_1005    conda-forge\n",
    "r-gargle                  1.5.2             r43h785f33e_1    conda-forge\n",
    "r-generics                0.1.3             r43hc72bb7e_3    conda-forge\n",
    "r-getopt                  1.20.4            r43ha770c72_1    conda-forge\n",
    "r-ggplot2                 3.5.1             r43hc72bb7e_1    conda-forge\n",
    "r-glue                    1.8.0             r43h2b5f3a1_0    conda-forge\n",
    "r-googledrive             2.1.1             r43hc72bb7e_2    conda-forge\n",
    "r-googlesheets4           1.1.1             r43h785f33e_2    conda-forge\n",
    "r-gtable                  0.3.6             r43hc72bb7e_0    conda-forge\n",
    "r-haven                   2.5.4             r43h0d4f4ea_1    conda-forge\n",
    "r-highr                   0.11              r43hc72bb7e_1    conda-forge\n",
    "r-hms                     1.1.3             r43hc72bb7e_2    conda-forge\n",
    "r-htmltools               0.5.8.1           r43ha18555a_1    conda-forge\n",
    "r-httr                    1.4.7             r43hc72bb7e_1    conda-forge\n",
    "r-ids                     1.0.1             r43hc72bb7e_4    conda-forge\n",
    "r-invgamma                1.1               r43hc72bb7e_4    conda-forge\n",
    "r-irlba                   2.3.5.1           r43h0d28552_3    conda-forge\n",
    "r-isoband                 0.2.7             r43ha18555a_3    conda-forge\n",
    "r-jquerylib               0.1.4             r43hc72bb7e_3    conda-forge\n",
    "r-jsonlite                1.8.9             r43h2b5f3a1_0    conda-forge\n",
    "r-knitr                   1.49              r43hc72bb7e_0    conda-forge\n",
    "r-labeling                0.4.3             r43hc72bb7e_1    conda-forge\n",
    "r-lambda.r                1.2.4             r43hc72bb7e_4    conda-forge\n",
    "r-languageserver          0.3.16            r43hdb488b9_1    conda-forge\n",
    "r-lattice                 0.22_6            r43hb1dbf0f_1    conda-forge\n",
    "r-lazyeval                0.2.2             r43hb1dbf0f_5    conda-forge\n",
    "r-lifecycle               1.0.4             r43hc72bb7e_1    conda-forge\n",
    "r-lintr                   3.1.2             r43hc72bb7e_1    conda-forge\n",
    "r-locfit                  1.5_9.10          r43h2b5f3a1_0    conda-forge\n",
    "r-lubridate               1.9.4             r43h2b5f3a1_0    conda-forge\n",
    "r-magrittr                2.0.3             r43hb1dbf0f_3    conda-forge\n",
    "r-mass                    7.3_60.0.1        r43hb1dbf0f_1    conda-forge\n",
    "r-matrix                  1.6_5             r43he966344_1    conda-forge\n",
    "r-matrixstats             1.4.1             r43h2b5f3a1_0    conda-forge\n",
    "r-memoise                 2.0.1             r43hc72bb7e_3    conda-forge\n",
    "r-mgcv                    1.9_1             r43h0d28552_1    conda-forge\n",
    "r-mime                    0.12              r43hb1dbf0f_3    conda-forge\n",
    "r-mixsqp                  0.3_54            r43hb79369c_2    conda-forge\n",
    "r-modelr                  0.1.11            r43hc72bb7e_2    conda-forge\n",
    "r-munsell                 0.5.1             r43hc72bb7e_1    conda-forge\n",
    "r-mvtnorm                 1.3_2             r43h9ad1c49_0    conda-forge\n",
    "r-nlme                    3.1_165           r43hbcb9c34_1    conda-forge\n",
    "r-numderiv                2016.8_1.1        r43hc72bb7e_6    conda-forge\n",
    "r-openssl                 2.2.2             r43he8289e2_0    conda-forge\n",
    "r-optparse                1.7.5             r43hc72bb7e_1    conda-forge\n",
    "r-pillar                  1.9.0             r43hc72bb7e_2    conda-forge\n",
    "r-pkgbuild                1.4.5             r43hc72bb7e_0    conda-forge\n",
    "r-pkgconfig               2.0.3             r43hc72bb7e_4    conda-forge\n",
    "r-pkgload                 1.4.0             r43hc72bb7e_0    conda-forge\n",
    "r-plyr                    1.8.9             r43ha18555a_1    conda-forge\n",
    "r-prettyunits             1.2.0             r43hc72bb7e_1    conda-forge\n",
    "r-processx                3.8.4             r43hb1dbf0f_1    conda-forge\n",
    "r-progress                1.2.3             r43hc72bb7e_1    conda-forge\n",
    "r-ps                      1.8.1             r43h2b5f3a1_0    conda-forge\n",
    "r-purrr                   1.0.2             r43hdb488b9_1    conda-forge\n",
    "r-r.cache                 0.16.0            r43hc72bb7e_3    conda-forge\n",
    "r-r.methodss3             1.8.2             r43hc72bb7e_3    conda-forge\n",
    "r-r.oo                    1.27.0            r43hc72bb7e_0    conda-forge\n",
    "r-r.utils                 2.12.3            r43hc72bb7e_1    conda-forge\n",
    "r-r6                      2.5.1             r43hc72bb7e_3    conda-forge\n",
    "r-ragg                    1.3.3             r43h9aa3752_0    conda-forge\n",
    "r-rappdirs                0.3.3             r43hb1dbf0f_3    conda-forge\n",
    "r-rcolorbrewer            1.1_3             r43h785f33e_3    conda-forge\n",
    "r-rcpp                    1.0.13_1          r43h93ab643_0    conda-forge\n",
    "r-rcpparmadillo           14.2.2_1          r43hb79369c_0    conda-forge\n",
    "r-rcppeigen               0.3.4.0.2         r43hb79369c_0    conda-forge\n",
    "r-rcppnumerical           0.6_0             r43h0d4f4ea_1    conda-forge\n",
    "r-rcurl                   1.98_1.16         r43he8228da_1    conda-forge\n",
    "r-readr                   2.1.5             r43h0d4f4ea_1    conda-forge\n",
    "r-readxl                  1.4.3             r43he58e087_1    conda-forge\n",
    "r-rematch                 2.0.0             r43hc72bb7e_1    conda-forge\n",
    "r-rematch2                2.1.2             r43hc72bb7e_4    conda-forge\n",
    "r-remotes                 2.5.0             r43hc72bb7e_1    conda-forge\n",
    "r-repr                    1.1.7             r43h785f33e_1    conda-forge\n",
    "r-reprex                  2.1.1             r43hc72bb7e_1    conda-forge\n",
    "r-restfulr                0.0.15            r43h56115f1_4    bioconda\n",
    "r-rex                     1.2.1             r43hc72bb7e_3    conda-forge\n",
    "r-rjson                   0.2.23            r43h93ab643_0    conda-forge\n",
    "r-rlang                   1.1.4             r43ha18555a_1    conda-forge\n",
    "r-rmarkdown               2.29              r43hc72bb7e_0    conda-forge\n",
    "r-roxygen2                7.3.2             r43h0d4f4ea_1    conda-forge\n",
    "r-rprojroot               2.0.4             r43hc72bb7e_1    conda-forge\n",
    "r-rstudioapi              0.17.1            r43hc72bb7e_0    conda-forge\n",
    "r-rvest                   1.0.4             r43hc72bb7e_1    conda-forge\n",
    "r-sass                    0.4.9             r43ha18555a_1    conda-forge\n",
    "r-scales                  1.3.0             r43hc72bb7e_1    conda-forge\n",
    "r-selectr                 0.4_2             r43hc72bb7e_4    conda-forge\n",
    "r-snow                    0.4_4             r43hc72bb7e_3    conda-forge\n",
    "r-squarem                 2021.1            r43hc72bb7e_3    conda-forge\n",
    "r-stringi                 1.8.4             r43h33cde33_3    conda-forge\n",
    "r-stringr                 1.5.1             r43h785f33e_1    conda-forge\n",
    "r-styler                  1.10.3            r43hc72bb7e_1    conda-forge\n",
    "r-sys                     3.4.3             r43h2b5f3a1_0    conda-forge\n",
    "r-systemfonts             1.1.0             r43h38d38ca_1    conda-forge\n",
    "r-textshaping             0.4.0             r43ha47bcaa_2    conda-forge\n",
    "r-tibble                  3.2.1             r43hdb488b9_3    conda-forge\n",
    "r-tidyr                   1.3.1             r43h0d4f4ea_1    conda-forge\n",
    "r-tidyselect              1.2.1             r43hc72bb7e_1    conda-forge\n",
    "r-tidyverse               2.0.0             r43h785f33e_2    conda-forge\n",
    "r-timechange              0.3.0             r43ha18555a_1    conda-forge\n",
    "r-tinytex                 0.54              r43hc72bb7e_0    conda-forge\n",
    "r-truncnorm               1.0_9             r43hdb488b9_2    conda-forge\n",
    "r-tzdb                    0.4.0             r43ha18555a_2    conda-forge\n",
    "r-utf8                    1.2.4             r43hb1dbf0f_1    conda-forge\n",
    "r-uuid                    1.2_1             r43hdb488b9_0    conda-forge\n",
    "r-vctrs                   0.6.5             r43h0d4f4ea_1    conda-forge\n",
    "r-viridislite             0.4.2             r43hc72bb7e_2    conda-forge\n",
    "r-vroom                   1.6.5             r43h0d4f4ea_1    conda-forge\n",
    "r-withr                   3.0.2             r43hc72bb7e_0    conda-forge\n",
    "r-xfun                    0.49              r43h93ab643_0    conda-forge\n",
    "r-xml                     3.99_0.17         r43h5bae778_2    conda-forge\n",
    "r-xml2                    1.3.6             r43h8194278_2    conda-forge\n",
    "r-xmlparsedata            1.0.5             r43hc72bb7e_3    conda-forge\n",
    "r-yaml                    2.3.10            r43hdb488b9_0    conda-forge\n",
    "readline                  8.2                  h8228510_1    conda-forge\n",
    "referencing               0.35.1             pyhd8ed1ab_1    conda-forge\n",
    "requests                  2.32.3             pyhd8ed1ab_1    conda-forge\n",
    "reretry                   0.11.8             pyhd8ed1ab_0    conda-forge\n",
    "rich                      13.9.4             pyhd8ed1ab_1    conda-forge\n",
    "rich-click                1.8.3              pyhd8ed1ab_0    conda-forge\n",
    "rpds-py                   0.22.3          py312h12e396e_0    conda-forge\n",
    "samtools                  1.21                 h50ea8bc_0    bioconda\n",
    "sed                       4.8                  he412f7d_0    conda-forge\n",
    "setuptools                75.6.0             pyhff2d567_1    conda-forge\n",
    "shellingham               1.5.4              pyhd8ed1ab_1    conda-forge\n",
    "six                       1.17.0             pyhd8ed1ab_0    conda-forge\n",
    "slack-sdk                 3.33.5             pyhd8ed1ab_0    conda-forge\n",
    "slack_sdk                 3.33.5               hd8ed1ab_0    conda-forge\n",
    "smart_open                7.0.5              pyhd8ed1ab_1    conda-forge\n",
    "smmap                     5.0.0              pyhd8ed1ab_0    conda-forge\n",
    "snakemake                 8.25.5               hdfd78af_0    bioconda\n",
    "snakemake-executor-plugin-cluster-generic 1.0.9              pyhdfd78af_0    bioconda\n",
    "snakemake-interface-common 1.17.4             pyhdfd78af_0    bioconda\n",
    "snakemake-interface-executor-plugins 9.3.2              pyhdfd78af_0    bioconda\n",
    "snakemake-interface-report-plugins 1.1.0              pyhdfd78af_0    bioconda\n",
    "snakemake-interface-storage-plugins 3.3.0              pyhdfd78af_0    bioconda\n",
    "snakemake-minimal         8.25.5             pyhdfd78af_0    bioconda\n",
    "spectra                   0.0.11                     py_1    conda-forge\n",
    "sqlite                    3.47.2               h9eae976_0    conda-forge\n",
    "stack_data                0.6.3              pyhd8ed1ab_1    conda-forge\n",
    "star                      2.7.11b              h43eeafb_3    bioconda\n",
    "subread                   2.0.8                h577a1d6_0    bioconda\n",
    "sysroot_linux-64          2.17                h4a8ded7_18    conda-forge\n",
    "tabulate                  0.9.0              pyhd8ed1ab_2    conda-forge\n",
    "tenacity                  9.0.0              pyhd8ed1ab_1    conda-forge\n",
    "throttler                 1.2.2              pyhd8ed1ab_0    conda-forge\n",
    "tk                        8.6.13          noxft_h4845f30_101    conda-forge\n",
    "tktable                   2.10                 h8bc8fbc_6    conda-forge\n",
    "toml                      0.10.2             pyhd8ed1ab_1    conda-forge\n",
    "tomli                     2.2.1              pyhd8ed1ab_1    conda-forge\n",
    "tomlkit                   0.13.2             pyha770c72_1    conda-forge\n",
    "tornado                   6.4.2           py312h66e93f0_0    conda-forge\n",
    "tqdm                      4.67.1             pyhd8ed1ab_0    conda-forge\n",
    "traitlets                 5.14.3             pyhd8ed1ab_1    conda-forge\n",
    "typeguard                 4.4.1              pyhd8ed1ab_1    conda-forge\n",
    "typer                     0.15.1             pyhd8ed1ab_0    conda-forge\n",
    "typer-slim                0.15.1             pyhd8ed1ab_0    conda-forge\n",
    "typer-slim-standard       0.15.1               hd8ed1ab_0    conda-forge\n",
    "typing-extensions         4.12.2               hd8ed1ab_1    conda-forge\n",
    "typing_extensions         4.12.2             pyha770c72_1    conda-forge\n",
    "tzdata                    2024b                hc8b5060_0    conda-forge\n",
    "ubiquerg                  0.8.0              pyhd8ed1ab_0    conda-forge\n",
    "urllib3                   2.2.3              pyhd8ed1ab_1    conda-forge\n",
    "veracitools               0.1.3                      py_0    conda-forge\n",
    "wcwidth                   0.2.13             pyhd8ed1ab_1    conda-forge\n",
    "wheel                     0.45.1             pyhd8ed1ab_1    conda-forge\n",
    "wrapt                     1.17.0          py312h66e93f0_0    conda-forge\n",
    "xmltodict                 0.14.2             pyhd8ed1ab_1    conda-forge\n",
    "xopen                     2.0.2              pyh707e725_2    conda-forge\n",
    "xorg-libice               1.1.1                hb9d3cd8_1    conda-forge\n",
    "xorg-libsm                1.2.4                he73a12e_1    conda-forge\n",
    "xorg-libx11               1.8.10               h4f16b4b_1    conda-forge\n",
    "xorg-libxau               1.0.11               hb9d3cd8_1    conda-forge\n",
    "xorg-libxdmcp             1.1.5                hb9d3cd8_0    conda-forge\n",
    "xorg-libxext              1.3.6                hb9d3cd8_0    conda-forge\n",
    "xorg-libxfixes            6.0.1                hb9d3cd8_0    conda-forge\n",
    "xorg-libxi                1.8.2                hb9d3cd8_0    conda-forge\n",
    "xorg-libxrandr            1.5.4                hb9d3cd8_0    conda-forge\n",
    "xorg-libxrender           0.9.11               hb9d3cd8_2    conda-forge\n",
    "xorg-libxt                1.3.1                hb9d3cd8_0    conda-forge\n",
    "xorg-libxtst              1.2.5                hb9d3cd8_3    conda-forge\n",
    "xz                        5.6.3                hbcc6ac9_1    conda-forge\n",
    "xz-gpl-tools              5.6.3                hbcc6ac9_1    conda-forge\n",
    "xz-tools                  5.6.3                hb9d3cd8_1    conda-forge\n",
    "yaml                      0.2.5                h7f98852_2    conda-forge\n",
    "yq                        3.4.3              pyhd8ed1ab_1    conda-forge\n",
    "yte                       1.5.5              pyha770c72_0    conda-forge\n",
    "zeromq                    4.3.5                h3b0a872_7    conda-forge\n",
    "zipp                      3.21.0             pyhd8ed1ab_1    conda-forge\n",
    "zlib-ng                   2.2.2                h5888daf_0    conda-forge\n",
    "zstandard                 0.23.0          py312hef9b889_1    conda-forge\n",
    "zstd                      1.5.6                ha6fb4c9_0    conda-forge\n",
    "(final_project) [mccallke0364@cn14 Bioinformatics-UWEC-Coursework-Liver-RNA-Seq]$ "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "_libgcc_mutex             0.1                 conda_forge    conda-forge\n",
    "_openmp_mutex             4.5                       2_gnu    conda-forge\n",
    "amply                     0.1.6              pyhd8ed1ab_0    conda-forge\n",
    "annotated-types           0.7.0              pyhd8ed1ab_0    conda-forge\n",
    "appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge\n",
    "argparse-dataclass        2.0.0              pyhd8ed1ab_0    conda-forge\n",
    "asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge\n",
    "attrs                     24.2.0             pyh71513ae_0    conda-forge\n",
    "brotli-python             1.1.0           py312h2ec8cdc_2    conda-forge\n",
    "bzip2                     1.0.8                h4bc722e_7    conda-forge\n",
    "ca-certificates           2024.11.26           h06a4308_0  \n",
    "certifi                   2024.8.30       py312h06a4308_0  \n",
    "cffi                      1.17.1          py312h06ac9bb_0    conda-forge\n",
    "charset-normalizer        3.4.0              pyhd8ed1ab_0    conda-forge\n",
    "click                     8.1.7           unix_pyh707e725_0    conda-forge\n",
    "coin-or-cbc               2.10.12              h8b142ea_1    conda-forge\n",
    "coin-or-cgl               0.60.9               h1d3f3f2_0    conda-forge\n",
    "coin-or-clp               1.17.10              h07f2a63_0    conda-forge\n",
    "coin-or-osi               0.108.11             h6514dde_1    conda-forge\n",
    "coin-or-utils             2.11.12              h99da652_1    conda-forge\n",
    "coincbc                   2.10.12           1_metapackage    conda-forge\n",
    "colorama                  0.4.6              pyhd8ed1ab_0    conda-forge\n",
    "coloredlogs               15.0.1             pyhd8ed1ab_3    conda-forge\n",
    "colormath                 3.0.0                      py_2    conda-forge\n",
    "comm                      0.2.2              pyhd8ed1ab_0    conda-forge\n",
    "conda-inject              1.3.2              pyhd8ed1ab_0    conda-forge\n",
    "configargparse            1.7                pyhd8ed1ab_0    conda-forge\n",
    "connection_pool           0.0.3              pyhd3deb0d_0    conda-forge\n",
    "cutadapt                  4.9             py312hf67a6ed_2    bioconda\n",
    "datrie                    0.8.2           py312h66e93f0_8    conda-forge\n",
    "debugpy                   1.8.8           py312h2ec8cdc_0    conda-forge\n",
    "decorator                 5.1.1              pyhd8ed1ab_0    conda-forge\n",
    "dnaio                     1.2.2           py312hf67a6ed_0    bioconda\n",
    "docutils                  0.21.2             pyhd8ed1ab_0    conda-forge\n",
    "dpath                     2.2.0              pyha770c72_0    conda-forge\n",
    "eido                      0.2.4              pyhd8ed1ab_0    conda-forge\n",
    "exceptiongroup            1.2.2              pyhd8ed1ab_0    conda-forge\n",
    "executing                 2.1.0              pyhd8ed1ab_0    conda-forge\n",
    "expat                     2.6.4                h5888daf_0    conda-forge\n",
    "fastqc                    0.12.1               hdfd78af_0    bioconda\n",
    "font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge\n",
    "font-ttf-inconsolata      3.000                h77eed37_0    conda-forge\n",
    "font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge\n",
    "font-ttf-ubuntu           0.83                 h77eed37_3    conda-forge\n",
    "fontconfig                2.15.0               h7e30c49_1    conda-forge\n",
    "fonts-conda-forge         1                             0    conda-forge\n",
    "freetype                  2.12.1               h267a509_2    conda-forge\n",
    "gdbm                      1.18                 hd4cb3f1_4  \n",
    "gitdb                     4.0.11             pyhd8ed1ab_0    conda-forge\n",
    "gitpython                 3.1.43             pyhd8ed1ab_0    conda-forge\n",
    "h2                        4.1.0              pyhd8ed1ab_0    conda-forge\n",
    "hpack                     4.0.0              pyh9f0ad1d_0    conda-forge\n",
    "humanfriendly             10.0               pyhd81877a_7    conda-forge\n",
    "humanize                  4.11.0             pyhd8ed1ab_1    conda-forge\n",
    "hyperframe                6.0.1              pyhd8ed1ab_0    conda-forge\n",
    "idna                      3.10               pyhd8ed1ab_0    conda-forge\n",
    "immutables                0.21            py312h66e93f0_0    conda-forge\n",
    "importlib-metadata        8.5.0              pyha770c72_0    conda-forge\n",
    "importlib_resources       6.4.5              pyhd8ed1ab_0    conda-forge\n",
    "iniconfig                 2.0.0              pyhd8ed1ab_0    conda-forge\n",
    "ipykernel                 6.29.5             pyh3099207_0    conda-forge\n",
    "ipython                   8.29.0             pyh707e725_0    conda-forge\n",
    "isa-l                     2.31.0               h4bc722e_2    conda-forge\n",
    "jedi                      0.19.2             pyhff2d567_0    conda-forge\n",
    "jinja2                    3.1.4              pyhd8ed1ab_0    conda-forge\n",
    "jsonschema                4.23.0             pyhd8ed1ab_0    conda-forge\n",
    "jsonschema-specifications 2024.10.1          pyhd8ed1ab_0    conda-forge\n",
    "jupyter_client            8.6.3              pyhd8ed1ab_0    conda-forge\n",
    "jupyter_core              5.7.2              pyh31011fe_1    conda-forge\n",
    "kaleido-core              0.2.1                h3644ca4_0    conda-forge\n",
    "keyutils                  1.6.1                h166bdaf_0    conda-forge\n",
    "krb5                      1.21.3               h659f571_0    conda-forge\n",
    "lcms2                     2.16                 hb7c19ff_0    conda-forge\n",
    "ld_impl_linux-64          2.43                 h712a8e2_2    conda-forge\n",
    "lerc                      4.0.0                h27087fc_0    conda-forge\n",
    "libblas                   3.9.0           25_linux64_openblas    conda-forge\n",
    "libcblas                  3.9.0           25_linux64_openblas    conda-forge\n",
    "libdeflate                1.22                 hb9d3cd8_0    conda-forge\n",
    "libedit                   3.1.20191231         he28a2e2_2    conda-forge\n",
    "libexpat                  2.6.4                h5888daf_0    conda-forge\n",
    "libffi                    3.4.2                h7f98852_5    conda-forge\n",
    "libgcc                    14.2.0               h77fa898_1    conda-forge\n",
    "libgcc-ng                 14.2.0               h69a702a_1    conda-forge\n",
    "libgfortran               14.2.0               h69a702a_1    conda-forge\n",
    "libgfortran-ng            14.2.0               h69a702a_1    conda-forge\n",
    "libgfortran5              14.2.0               hd5240d6_1    conda-forge\n",
    "libgomp                   14.2.0               h77fa898_1    conda-forge\n",
    "libjpeg-turbo             3.0.0                hd590300_1    conda-forge\n",
    "liblapack                 3.9.0           25_linux64_openblas    conda-forge\n",
    "liblapacke                3.9.0           25_linux64_openblas    conda-forge\n",
    "liblzma                   5.6.3                hb9d3cd8_1    conda-forge\n",
    "libnsl                    2.0.1                hd590300_0    conda-forge\n",
    "libopenblas               0.3.28          pthreads_h94d23a6_1    conda-forge\n",
    "libpng                    1.6.44               hadc24fc_0    conda-forge\n",
    "libsodium                 1.0.20               h4ab18f5_0    conda-forge\n",
    "libsqlite                 3.47.0               hadc24fc_1    conda-forge\n",
    "libstdcxx                 14.2.0               hc0a3c3a_1    conda-forge\n",
    "libstdcxx-ng              14.2.0               h4852527_1    conda-forge\n",
    "libtiff                   4.7.0                hc4654cb_2    conda-forge\n",
    "libuuid                   2.38.1               h0b41bf4_0    conda-forge\n",
    "libwebp-base              1.4.0                hd590300_0    conda-forge\n",
    "libxcb                    1.17.0               h8a09558_0    conda-forge\n",
    "libxcrypt                 4.4.36               hd590300_1    conda-forge\n",
    "libzlib                   1.3.1                hb9d3cd8_2    conda-forge\n",
    "logmuse                   0.2.8              pyhd8ed1ab_0    conda-forge\n",
    "markdown                  3.6                pyhd8ed1ab_0    conda-forge\n",
    "markdown-it-py            3.0.0              pyhd8ed1ab_0    conda-forge\n",
    "markupsafe                3.0.2           py312h178313f_0    conda-forge\n",
    "mathjax                   2.7.7                ha770c72_3    conda-forge\n",
    "matplotlib-inline         0.1.7              pyhd8ed1ab_0    conda-forge\n",
    "mdurl                     0.1.2              pyhd8ed1ab_0    conda-forge\n",
    "multiqc                   1.25.2             pyhdfd78af_0    bioconda\n",
    "natsort                   8.4.0              pyh29332c3_1    conda-forge\n",
    "nbformat                  5.10.4             pyhd8ed1ab_0    conda-forge\n",
    "ncurses                   6.5                  he02047a_1    conda-forge\n",
    "nest-asyncio              1.6.0              pyhd8ed1ab_0    conda-forge\n",
    "networkx                  3.4.2              pyh267e887_2    conda-forge\n",
    "nspr                      4.36                 h5888daf_0    conda-forge\n",
    "nss                       3.107                hdf54f9c_0    conda-forge\n",
    "numpy                     2.1.3           py312h58c1407_0    conda-forge\n",
    "openjdk                   8.0.152              h7b6447c_3  \n",
    "openjpeg                  2.5.3                h5fbd93e_0    conda-forge\n",
    "openssl                   3.4.0                hb9d3cd8_0    conda-forge\n",
    "packaging                 24.2               pyhff2d567_1    conda-forge\n",
    "pandas                    2.2.2           py312h1d6d2e6_1    conda-forge\n",
    "parso                     0.8.4              pyhd8ed1ab_0    conda-forge\n",
    "pbzip2                    1.1.13               h1fcc475_2    conda-forge\n",
    "pephubclient              0.4.4              pyhd8ed1ab_0    conda-forge\n",
    "peppy                     0.40.7             pyhd8ed1ab_1    conda-forge\n",
    "perl                      5.34.0               h5eee18b_2  \n",
    "pexpect                   4.9.0              pyhd8ed1ab_0    conda-forge\n",
    "pickleshare               0.7.5                   py_1003    conda-forge\n",
    "pigz                      2.8                  h2797004_0    conda-forge\n",
    "pillow                    11.0.0          py312h7b63e92_0    conda-forge\n",
    "pip                       24.3.1             pyh8b19718_0    conda-forge\n",
    "pkgutil-resolve-name      1.3.10             pyhd8ed1ab_1    conda-forge\n",
    "plac                      1.4.3              pyhd8ed1ab_0    conda-forge\n",
    "platformdirs              4.3.6              pyhd8ed1ab_0    conda-forge\n",
    "plotly                    5.24.1             pyhd8ed1ab_1    conda-forge\n",
    "pluggy                    1.5.0              pyhd8ed1ab_0    conda-forge\n",
    "prompt-toolkit            3.0.48             pyha770c72_0    conda-forge\n",
    "psutil                    6.1.0           py312h66e93f0_0    conda-forge\n",
    "pthread-stubs             0.4               hb9d3cd8_1002    conda-forge\n",
    "ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge\n",
    "pulp                      2.8.0           py312h7900ff3_0    conda-forge\n",
    "pure_eval                 0.2.3              pyhd8ed1ab_0    conda-forge\n",
    "pyaml-env                 1.2.1              pyhd8ed1ab_0    conda-forge\n",
    "pycparser                 2.22               pyhd8ed1ab_0    conda-forge\n",
    "pydantic                  2.9.2              pyhd8ed1ab_0    conda-forge\n",
    "pydantic-core             2.23.4          py312h12e396e_0    conda-forge\n",
    "pygments                  2.18.0             pyhd8ed1ab_0    conda-forge\n",
    "pyparsing                 3.2.0              pyhd8ed1ab_1    conda-forge\n",
    "pysocks                   1.7.1              pyha2e5f31_6    conda-forge\n",
    "pytest                    8.3.3              pyhd8ed1ab_0    conda-forge\n",
    "python                    3.12.8          h9e4cc4f_1_cpython    conda-forge\n",
    "python-dateutil           2.9.0.post0        pyhff2d567_0    conda-forge\n",
    "python-fastjsonschema     2.20.0             pyhd8ed1ab_0    conda-forge\n",
    "python-isal               1.7.1           py312h66e93f0_0    conda-forge\n",
    "python-kaleido            0.2.1              pyhd8ed1ab_0    conda-forge\n",
    "python-tzdata             2024.2             pyhd8ed1ab_0    conda-forge\n",
    "python-zlib-ng            0.5.1           py312h27ca174_0    conda-forge\n",
    "python_abi                3.12                    5_cp312    conda-forge\n",
    "pytz                      2024.2             pyhd8ed1ab_0    conda-forge\n",
    "pyyaml                    6.0.2           py312h66e93f0_1    conda-forge\n",
    "pyzmq                     26.2.0          py312hbf22597_3    conda-forge\n",
    "readline                  8.2                  h8228510_1    conda-forge\n",
    "referencing               0.35.1             pyhd8ed1ab_0    conda-forge\n",
    "requests                  2.32.3             pyhd8ed1ab_0    conda-forge\n",
    "reretry                   0.11.8             pyhd8ed1ab_0    conda-forge\n",
    "rich                      13.9.4             pyhd8ed1ab_0    conda-forge\n",
    "rich-click                1.8.3              pyhd8ed1ab_0    conda-forge\n",
    "rpds-py                   0.21.0          py312h12e396e_0    conda-forge\n",
    "setuptools                75.5.0             pyhff2d567_0    conda-forge\n",
    "shellingham               1.5.4              pyhd8ed1ab_0    conda-forge\n",
    "six                       1.16.0             pyh6c4a22f_0    conda-forge\n",
    "slack-sdk                 3.33.4             pyhff2d567_1    conda-forge\n",
    "slack_sdk                 3.33.4               hff2d567_1    conda-forge\n",
    "smart_open                7.0.5              pyhd8ed1ab_1    conda-forge\n",
    "smmap                     5.0.0              pyhd8ed1ab_0    conda-forge\n",
    "snakemake                 8.25.3               hdfd78af_0    bioconda\n",
    "snakemake-interface-common 1.17.4             pyhdfd78af_0    bioconda\n",
    "snakemake-interface-executor-plugins 9.3.2              pyhdfd78af_0    bioconda\n",
    "snakemake-interface-report-plugins 1.1.0              pyhdfd78af_0    bioconda\n",
    "snakemake-interface-storage-plugins 3.3.0              pyhdfd78af_0    bioconda\n",
    "snakemake-minimal         8.25.3             pyhdfd78af_0    bioconda\n",
    "spectra                   0.0.11                     py_1    conda-forge\n",
    "sqlite                    3.47.0               h9eae976_1    conda-forge\n",
    "stack_data                0.6.2              pyhd8ed1ab_0    conda-forge\n",
    "star                      2.5.2b                        0    bioconda\n",
    "tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge\n",
    "tenacity                  9.0.0              pyhd8ed1ab_1    conda-forge\n",
    "throttler                 1.2.2              pyhd8ed1ab_0    conda-forge\n",
    "tk                        8.6.13          noxft_h4845f30_101    conda-forge\n",
    "tomli                     2.1.0              pyhff2d567_0    conda-forge\n",
    "tornado                   6.4.1           py312h66e93f0_1    conda-forge\n",
    "tqdm                      4.67.1             pyhd8ed1ab_0    conda-forge\n",
    "traitlets                 5.14.3             pyhd8ed1ab_0    conda-forge\n",
    "typeguard                 4.4.1              pyhd8ed1ab_1    conda-forge\n",
    "typer                     0.13.0             pyhd8ed1ab_0    conda-forge\n",
    "typer-slim                0.13.0             pyhff2d567_0    conda-forge\n",
    "typer-slim-standard       0.13.0               hd8ed1ab_0    conda-forge\n",
    "typing-extensions         4.12.2               hd8ed1ab_0    conda-forge\n",
    "typing_extensions         4.12.2             pyha770c72_0    conda-forge\n",
    "tzdata                    2024b                hc8b5060_0    conda-forge\n",
    "ubiquerg                  0.8.0              pyhd8ed1ab_0    conda-forge\n",
    "urllib3                   2.2.3              pyhd8ed1ab_0    conda-forge\n",
    "veracitools               0.1.3                      py_0    conda-forge\n",
    "wcwidth                   0.2.13             pyhd8ed1ab_0    conda-forge\n",
    "wheel                     0.45.0             pyhd8ed1ab_0    conda-forge\n",
    "wrapt                     1.16.0          py312h66e93f0_1    conda-forge\n",
    "xopen                     2.0.2              pyh707e725_1    conda-forge\n",
    "xorg-libxau               1.0.11               hb9d3cd8_1    conda-forge\n",
    "xorg-libxdmcp             1.1.5                hb9d3cd8_0    conda-forge\n",
    "xz                        5.2.6                h166bdaf_0    conda-forge\n",
    "yaml                      0.2.5                h7f98852_2    conda-forge\n",
    "yte                       1.5.4              pyha770c72_0    conda-forge\n",
    "zeromq                    4.3.5                h3b0a872_7    conda-forge\n",
    "zipp                      3.21.0             pyhd8ed1ab_0    conda-forge\n",
    "zlib-ng                   2.2.2                h5888daf_0    conda-forge\n",
    "zstandard                 0.23.0          py312hef9b889_1    conda-forge\n",
    "zstd                      1.5.6                ha6fb4c9_0    conda-forge"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
